diff -urN ntp-4.2.8.orig/scripts/build/mkver.in ntp-4.2.8/scripts/build/mkver.in
--- ntp-4.2.8.orig/scripts/build/mkver.in	2014-12-19 12:56:52.000000000 +0100
+++ ntp-4.2.8/scripts/build/mkver.in	2014-12-20 14:03:21.769320019 +0100
@@ -15,7 +15,7 @@
  *) ConfStr="${ConfStr}-@VER_SUFFIX@" ;;
 esac
 
-ConfStr="$ConfStr `LC_TIME=C TZ=UTC date`"
+#ConfStr="$ConfStr `LC_TIME=C TZ=UTC date`"
 
 if [ ! -f .version ]; then
   echo 0 > .version
