diff -urN ppp-2.4.6.orig/pppd/Makefile.linux ppp-2.4.6/pppd/Makefile.linux
--- ppp-2.4.6.orig/pppd/Makefile.linux	2014-06-18 10:37:26.773498437 +0200
+++ ppp-2.4.6/pppd/Makefile.linux	2014-06-18 10:38:17.243436695 +0200
@@ -124,10 +124,10 @@
 ifdef NO_CRYPT_HACK
 CFLAGS += -DNO_CRYPT_HACK
 else
-ifneq ($(wildcard /usr/include/crypt.h),)
+#ifneq ($(wildcard /usr/include/crypt.h),)
 CFLAGS  += -DHAVE_CRYPT_H=1
 LIBS	+= -lcrypt
-endif
+#endif
 endif
 
 ifdef USE_LIBUTIL
