diff -Nrdup dhcpdump-1.8.orig/dhcpdump.c dhcpdump-1.8.patched/dhcpdump.c
--- dhcpdump-1.8.orig/dhcpdump.c	2016-01-06 13:14:52.327400000 +0100
+++ dhcpdump-1.8.patched/dhcpdump.c	2016-01-06 15:50:43.983078339 +0100
@@ -36,6 +36,7 @@
 #define SPERM	(60)
 
 #define LARGESTRING 1024
+#define ARRAY_SIZE(x)  (sizeof(x) / sizeof((x)[0]))
 
 // header variables
 u_char	timestamp[40];			// timestamp on header
@@ -470,7 +471,7 @@ int printdata(u_char *data, int data_len
 	case 36:	// Ethernet encapsulation
 		printf("%d (%s)",
 		    data[j + 2],
-		    data[j +2 ] > sizeof(ethernet_encapsulation) ?
+		    data[j +2 ] > ARRAY_SIZE(ethernet_encapsulation) ?
 			"*wrong value*" :
 			ethernet_encapsulation[data[j + 2]]);
 		break;
@@ -478,7 +479,7 @@ int printdata(u_char *data, int data_len
 	case 52:	// Option overload
 		printf("%d (%s)",
 		    data[j + 2],
-		    data[j + 2] > sizeof(option_overload) ?
+		    data[j + 2] > ARRAY_SIZE(option_overload) ?
 			"*wrong value*" :
 			option_overload[data[j + 2]]);
 		break;
@@ -486,7 +487,7 @@ int printdata(u_char *data, int data_len
 	case 53:	// DHCP message type
 		printf("%d (%s)",
 		    data[j + 2],
-		    data[j + 2] > sizeof(dhcp_message_types) ?
+		    data[j + 2] > ARRAY_SIZE(dhcp_message_types) ?
 			"*wrong value*" :
 			dhcp_message_types[data[j + 2]]);
 		break;
@@ -519,7 +520,7 @@ int printdata(u_char *data, int data_len
 		printf("\n");
 		for (i = j + 2; i < j + data[j + 1]; ) {
 			printf("%-17s %-13s ", " ",
-			    data[i] > sizeof(relayagent_suboptions) ?
+			    data[i] > ARRAY_SIZE(relayagent_suboptions) ?
 			    "*wrong value*" :
 			    relayagent_suboptions[data[i]]);
 			if (i + data[i + 1] > j + data[j + 1]) {
