diff -urN wget-1.16.1.orig/src/Makefile.am wget-1.16.1/src/Makefile.am
--- wget-1.16.1.orig/src/Makefile.am	2014-12-07 20:52:24.000000000 +0100
+++ wget-1.16.1/src/Makefile.am	2015-03-24 23:23:41.553273515 +0100
@@ -76,9 +76,9 @@
 	echo '' >> $@
 	echo '#include "version.h"' >> $@
 	echo 'const char *version_string = "@VERSION@";' >> $@
-	echo 'const char *compilation_string = "'$(COMPILE)'";' \
+	echo 'const char *compilation_string = "'$(patsubst $(CC)%,$(notdir $(CC))%, $(COMPILE))'";' \
 	    | $(ESCAPEQUOTE) >> $@
-	echo 'const char *link_string = "'$(CCLD) $(AM_CFLAGS) $(CFLAGS) \
+	echo 'const char *link_string = "'$(notdir $(CCLD)) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) $(LIBS) $(wget_LDADD)'";' \
 	    | $(ESCAPEQUOTE) >> $@
 
diff -urN wget-1.16.1.orig/src/Makefile.in wget-1.16.1/src/Makefile.in
--- wget-1.16.1.orig/src/Makefile.in	2014-12-08 11:33:00.000000000 +0100
+++ wget-1.16.1/src/Makefile.in	2015-03-24 23:24:29.869704801 +0100
@@ -2209,9 +2209,9 @@
 	echo '' >> $@
 	echo '#include "version.h"' >> $@
 	echo 'const char *version_string = "@VERSION@";' >> $@
-	echo 'const char *compilation_string = "'$(COMPILE)'";' \
+	echo 'const char *compilation_string = "'$(patsubst $(CC)%,$(notdir $(CC))%, $(COMPILE))'";' \
 	    | $(ESCAPEQUOTE) >> $@
-	echo 'const char *link_string = "'$(CCLD) $(AM_CFLAGS) $(CFLAGS) \
+	echo 'const char *link_string = "'$(notdir $(CCLD)) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) $(LIBS) $(wget_LDADD)'";' \
 	    | $(ESCAPEQUOTE) >> $@
 
