diff -urN squid-3.2.3.orig/configure squid-3.2.3/configure
--- squid-3.2.3.orig/configure	2012-10-20 14:40:38.000000000 +0200
+++ squid-3.2.3/configure	2012-12-08 20:59:57.137211995 +0100
@@ -16809,7 +16809,7 @@
   done
 
 cat >>confdefs.h <<_ACEOF
-#define LT_DLSEARCH_PATH "$sys_dlsearch_path"
+#define LT_DLSEARCH_PATH "/lib:/usr/lib"
 _ACEOF
 
 fi
diff -urN squid-3.2.3.orig/libltdl/configure squid-3.2.3/libltdl/configure
--- squid-3.2.3.orig/libltdl/configure	2012-10-20 14:40:06.000000000 +0200
+++ squid-3.2.3/libltdl/configure	2012-12-08 20:59:57.205213380 +0100
@@ -10921,7 +10921,7 @@
   done
 
 cat >>confdefs.h <<_ACEOF
-#define LT_DLSEARCH_PATH "$sys_dlsearch_path"
+#define LT_DLSEARCH_PATH "/lib:/usr/lib"
 _ACEOF
 
 fi
