diff -urN gutenprint-5.2.9.orig/Makefile.am gutenprint-5.2.9/Makefile.am
--- gutenprint-5.2.9.orig/Makefile.am	2011-02-12 02:44:53.000000000 +0100
+++ gutenprint-5.2.9/Makefile.am	2012-08-14 19:46:02.588767617 +0200
@@ -27,7 +27,7 @@
 
 ## Variables
 
-SUBDIRS = include src samples test po man doc scripts
+SUBDIRS = include src samples test man doc scripts
 DISTDIRS = m4 m4extra
 
 docdir = $(datadir)/$(PACKAGE)/doc
diff -urN gutenprint-5.2.9.orig/scripts/autogen.sh gutenprint-5.2.9/scripts/autogen.sh
--- gutenprint-5.2.9.orig/scripts/autogen.sh	2011-12-20 15:05:27.000000000 +0100
+++ gutenprint-5.2.9/scripts/autogen.sh	2012-08-14 19:44:59.656768600 +0200
@@ -95,15 +95,6 @@
   echo "(or a newer version if it is available)"
 }
 
-(autopoint --version) < /dev/null > /dev/null 2>&1 || {
-  echo
-  echo "**Error**: You must have \`autopoint' installed to compile gutenprint."
-  echo "Get ftp://ftp.gnu.org/pub/gnu/gettext/gettext-0.11.5.tar.gz"
-  echo "(or a newer version if it is available)"
-  DIE=1
-  NO_AUTOMAKE=yes
-}
-
 (automake --version) < /dev/null > /dev/null 2>&1 || {
   echo
   echo "**Error**: You must have \`automake' installed to compile gutenprint."
@@ -342,8 +333,6 @@
 	  echo 'This ChangeLog is redundant. Please see the main ChangeLog for i18n changes.' > po/ChangeLog
 	  echo >> po/ChangeLog
 	  echo 'This file is present only to keep po/Makefile.in.in happy.' >> po/ChangeLog
-	  echo "Running autopoint...  Ignore non-fatal messages."
-	    autopoint --force
 	  echo "Making $dr/aclocal.m4 writable ..."
 	  test -r $dr/aclocal.m4 && chmod u+w $dr/aclocal.m4
         fi
diff -urN gutenprint-5.2.9.orig/src/Makefile.am gutenprint-5.2.9/src/Makefile.am
--- gutenprint-5.2.9.orig/src/Makefile.am	2008-06-14 04:45:39.000000000 +0200
+++ gutenprint-5.2.9/src/Makefile.am	2012-08-14 19:46:02.588767617 +0200
@@ -23,7 +23,6 @@
 
 SUBDIRS = 					\
 	main					\
-	xml					\
 	gutenprintui2				\
 	escputil				\
 	gimp2					\
