diff -urN libgpg-error-1.12.orig/src/version.c libgpg-error-1.12/src/version.c
--- libgpg-error-1.12.orig/src/version.c	2013-02-23 19:47:45.000000000 +0100
+++ libgpg-error-1.12/src/version.c	2013-11-20 20:26:09.636277214 +0100
@@ -41,7 +41,7 @@
     "This is Libgpg-error " PACKAGE_VERSION " - An error code library\n"
     "Copyright 2003, 2004, 2010, 2013 g10 Code GmbH\n"
     "\n"
-    "(" BUILD_REVISION " " BUILD_TIMESTAMP ")\n"
+    "(" BUILD_REVISION ")\n"
     "\n\n";
   return blurb;
 }
