diff -purN isdnutils-3.25.orig/capi20/convert.c isdnutils-3.25/capi20/convert.c
--- isdnutils-3.25.orig/capi20/convert.c	2012-07-06 08:27:19.000000000 +0200
+++ isdnutils-3.25/capi20/convert.c	2017-02-09 23:34:38.563903357 +0100
@@ -23,6 +23,14 @@
 
 char *capi_info2str(_cword reason)
 {
+    if ((reason & 0xff00) == 0x3400) {
+	/* mISDN does not set bit 7, which is correct as this bit does not
+	 * belong to the ETS 300 102-1/Q.931 cause value, cf. section 4.5.12
+	 * (ETS 300 102-1) or Table I.2 (Q.931)
+	 */
+	reason |= 0x80;
+    }
+
     switch (reason) {
 
 /*-- informative values (corresponding message was processed) -----*/
