diff -urN heimdal-7.1.0.orig/cf/db.m4 heimdal-7.1.0/cf/db.m4
--- heimdal-7.1.0.orig/cf/db.m4	2016-11-29 02:35:27.000000000 +0100
+++ heimdal-7.1.0/cf/db.m4	2017-08-30 20:49:54.372380951 +0200
@@ -188,7 +188,7 @@
   	  gdbm/ndbm.h				\
       ])
   
-      AC_FIND_FUNC_NO_LIBS(dbm_firstkey, gdbm, [
+      AC_FIND_FUNC_NO_LIBS(dbm_firstkey, gdbm_compat, [
       #include <stdio.h>
       #include <gdbm/ndbm.h>
       DBM *dbm;
