diff -urN isdnutils-3.9.20060704+dfsg.3.orig//vbox/src/Makefile.in isdnutils-3.9.20060704+dfsg.3/vbox/src/Makefile.in
--- isdnutils-3.9.20060704+dfsg.3.orig//vbox/src/Makefile.in	2012-02-27 18:35:17.398768235 +0100
+++ isdnutils-3.9.20060704+dfsg.3/vbox/src/Makefile.in	2012-02-28 17:02:24.898767223 +0100
@@ -81,7 +81,7 @@
 
 #----------------------------------------------------------------------------#
 
-all:		vboxlib vboxd vboxgetty vboxconvert vboxctrl vboxbeep vbox
+all:		vboxlib vboxd vboxgetty vboxconvert vboxctrl vboxbeep
 
 #----------------------------------------------------------------------------#
 # Compiling vboxlib                                                          #
