diff -urN isdnutils-3.9.20060704+dfsg.3.orig/vbox/src/Makefile.in isdnutils-3.9.20060704+dfsg.3.orig.patched/vbox/src/Makefile.in
--- isdnutils-3.9.20060704+dfsg.3.orig/vbox/src/Makefile.in	2012-04-05 11:24:26.000000000 +0200
+++ isdnutils-3.9.20060704+dfsg.3.orig.patched/vbox/src/Makefile.in	2012-04-05 11:25:50.000000000 +0200
@@ -101,7 +101,7 @@
 		  script.o voice.o lists.o rcvbox.o rcgetty.o \
 
 
-vboxgetty:	  $(VBOXGETTYOBJS)
+vboxgetty:	  $(VBOXGETTYOBJS) vboxlib
 		  $(CC) $(CFLAGS) $(LDFLAGS) $(VBOXGETTYOBJS) -o vboxgetty $(TCL_LIBS) -L. -lvbox
 
 #----------------------------------------------------------------------------#
@@ -110,7 +110,7 @@
 
 VBOXOBJS	= vbox.o streamio.o
 
-vbox:		  $(VBOXOBJS) libvbox.c libvbox.h
+vbox:		  $(VBOXOBJS) libvbox.c libvbox.h vboxlib
 		  $(CC) $(CFLAGS) $(LDFLAGS) $(VBOXOBJS) -o vbox $(NCURSES_LIBS) -L. -lvbox
 
 #----------------------------------------------------------------------------#
@@ -119,7 +119,7 @@
 
 VBOXCONVOBJS	= vboxconvert.o
 
-vboxconvert:	  $(VBOXCONVOBJS)
+vboxconvert:	  $(VBOXCONVOBJS) vboxlib
 		  $(CC) $(CFLAGS) $(LDFLAGS) $(VBOXCONVOBJS) -o vboxconvert -L. -lvbox
 
 #----------------------------------------------------------------------------#
@@ -128,7 +128,7 @@
 
 VBOXCTRLOBJS	= vboxctrl.o
 
-vboxctrl:	  $(VBOXCTRLOBJS)
+vboxctrl:	  $(VBOXCTRLOBJS) vboxlib
 		  $(CC) $(CFLAGS) $(LDFLAGS) $(VBOXCTRLOBJS) -o vboxctrl -L. -lvbox
 
 #----------------------------------------------------------------------------#
@@ -137,7 +137,7 @@
 
 VBOXBEEPOBJS	= vboxbeep.o
 
-vboxbeep:	  $(VBOXBEEPOBJS)
+vboxbeep:	  $(VBOXBEEPOBJS) vboxlib
 		  $(CC) $(CFLAGS) $(LDFLAGS) $(VBOXBEEPOBJS) -pie -o vboxbeep -L. -lvbox
 
 #----------------------------------------------------------------------------#
@@ -146,7 +146,7 @@
 
 VBOXDOBJS	= vboxd.o streamio.o
 
-vboxd:		  $(VBOXDOBJS)
+vboxd:		  $(VBOXDOBJS) vboxlib
 		  $(CC) $(CFLAGS) $(LDFLAGS) $(VBOXDOBJS) -o vboxd -L. -lvbox -lcrypt
 
 distclean: clean
