diff -purN oidentd-2.0.8.orig/src/oidentd_util.c oidentd-2.0.8/src/oidentd_util.c
--- oidentd-2.0.8.orig/src/oidentd_util.c	2006-05-22 02:31:19.000000000 +0200
+++ oidentd-2.0.8/src/oidentd_util.c	2015-12-02 00:42:25.941942343 +0100
@@ -75,7 +75,7 @@ int random_seed(void) {
 ** PRNG functions on systems whose libraries provide them.)
 */
 
-inline int randval(int i) {
+extern inline int randval(int i) {
 	/* Per _Numerical Recipes in C_: */
 	return ((double) i * rand() / (RAND_MAX+1.0));
 }
