diff -purN mini_httpd-1.29.orig/version.h mini_httpd-1.29/version.h
--- mini_httpd-1.29.orig/version.h	2018-05-24 01:26:50.000000000 +0200
+++ mini_httpd-1.29/version.h	2018-08-27 11:37:51.005017478 +0200
@@ -3,7 +3,7 @@
 #ifndef _VERSION_H_
 #define _VERSION_H_
 
-#define SERVER_SOFTWARE "mini_httpd/1.30 26Oct2018"
+#define SERVER_SOFTWARE "mini_httpd/1.30-fli4l 26Oct2018"
 #define SERVER_URL "http://www.acme.com/software/mini_httpd/"
 
 #endif /* _VERSION_H_ */
