diff -purN mini_httpd-1.27.orig/mini_httpd.c mini_httpd-1.27/mini_httpd.c
--- mini_httpd-1.27.orig/mini_httpd.c	2017-02-28 15:25:30.000000000 +0100
+++ mini_httpd-1.27/mini_httpd.c	2018-02-02 19:39:10.377259258 +0100
@@ -137,7 +137,7 @@ typedef long long int64_t;
 #define CGI_NICE 10
 #endif /* CGI_NICE */
 #ifndef CGI_PATH
-#define CGI_PATH "/usr/local/bin:/usr/ucb:/bin:/usr/bin"
+#define CGI_PATH "/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/bin:."
 #endif /* CGI_PATH */
 #ifndef CGI_LD_LIBRARY_PATH
 #define CGI_LD_LIBRARY_PATH "/usr/local/lib:/usr/lib"
