diff -purN mini_httpd-1.29.orig/mini_httpd.c mini_httpd-1.29/mini_httpd.c
--- mini_httpd-1.29.orig/mini_httpd.c	2018-08-27 11:30:59.757218869 +0200
+++ mini_httpd-1.29/mini_httpd.c	2018-08-27 11:32:46.562588335 +0200
@@ -744,6 +744,8 @@ main( int argc, char** argv )
 	    perror( "setuid" );
 	    exit( 1 );
 	    }
+#if 0 /* FFL-886: we need httpd to run as root as we have to run admin-only
+         tools from the CGIs */
 	/* Check for unnecessary security exposure. */
 	if ( ! do_chroot )
 	    {
@@ -752,6 +754,7 @@ main( int argc, char** argv )
 	    (void) fprintf( stderr,
 		"%s: started as root without requesting chroot(), warning only\n", argv0 );
 	    }
+#endif
 	}
 
     /* Catch various signals. */
