diff -purN mini_httpd-1.29.orig/mini_httpd.c mini_httpd-1.29/mini_httpd.c
--- mini_httpd-1.29.orig/mini_httpd.c	2018-08-27 11:33:54.746547525 +0200
+++ mini_httpd-1.29/mini_httpd.c	2018-08-27 11:34:05.698380357 +0200
@@ -2249,6 +2249,7 @@ make_envp( void )
     int envn;
     char* cp;
     char buf[256];
+    char rp[MAXPATHLEN];
 
     envn = 0;
     envp[envn++] = build_env( "PATH=%s", CGI_PATH );
@@ -2269,6 +2270,8 @@ make_envp( void )
     envp[envn++] = build_env(
 	"REQUEST_METHOD=%s", get_method_str( method ) );
     envp[envn++] = build_env( "SCRIPT_NAME=%s", path );
+    envp[envn++] = build_env( "SCRIPT_FILENAME=%s", realpath(file, rp) );
+    envp[envn++] = "DOCUMENT_ROOT=/usr/local/htdocs";
     if ( pathinfo != (char*) 0 )
 	{
 	envp[envn++] = build_env( "PATH_INFO=/%s", pathinfo );
