diff -purN mini_httpd-1.29.orig/mini_httpd.c mini_httpd-1.29/mini_httpd.c
--- mini_httpd-1.29.orig/mini_httpd.c	2018-08-27 11:35:24.505179816 +0200
+++ mini_httpd-1.29/mini_httpd.c	2018-08-27 11:35:32.521061558 +0200
@@ -1158,9 +1158,7 @@ handle_request( void )
     char* line;
     char* cp;
     int r, file_len, i;
-    const char* index_names[] = {
-	"index.html", "index.htm", "index.xhtml", "index.xht", "Default.htm",
-	"index.cgi" };
+    const char* index_names[] = { "index.cgi", "index.htm", "index.php", "index.html" };
 
     /* Set up the timeout for reading. */
 #ifdef HAVE_SIGSET
