diff -purN mini_httpd-1.29.orig/mini_httpd.c mini_httpd-1.29/mini_httpd.c
--- mini_httpd-1.29.orig/mini_httpd.c	2018-08-27 11:36:03.015611614 +0200
+++ mini_httpd-1.29/mini_httpd.c	2018-08-27 11:36:09.007523192 +0200
@@ -544,7 +544,7 @@ main( int argc, char** argv )
 	listen6_fd = initialize_listen_socket( &host_addr6 );
     else
 	listen6_fd = -1;
-    if ( gotv4 )
+    if ( gotv4 && listen6_fd == -1 )
 	listen4_fd = initialize_listen_socket( &host_addr4 );
     else
 	listen4_fd = -1;
