diff -urN rrdtool-1.4.8/src/rrd_cgi.c rrdtool-1.4.8.fli4l/src/rrd_cgi.c
--- rrdtool-1.4.8/src/rrd_cgi.c	2013-05-23 09:55:07.000000000 +0200
+++ rrdtool-1.4.8.fli4l/src/rrd_cgi.c	2013-11-23 16:09:39.000000000 +0100
@@ -709,7 +709,7 @@
         if (strcasecmp(args[0], "VERSION") == 0) {
             return stralloc(PACKAGE_VERSION);
         } else if (strcasecmp(args[0], "COMPILETIME") == 0) {
-            return stralloc(__DATE__ " " __TIME__);
+            return stralloc("(unknown)");
         } else {
             return stralloc("[ERROR: internal unknown argument]");
         }
diff -urN rrdtool-1.4.8/src/rrd_tool.c rrdtool-1.4.8.fli4l/src/rrd_tool.c
--- rrdtool-1.4.8/src/rrd_tool.c	2013-05-23 09:55:07.000000000 +0200
+++ rrdtool-1.4.8.fli4l/src/rrd_tool.c	2013-11-23 16:10:36.000000000 +0100
@@ -49,8 +49,7 @@
 
     const char *help_main =
         N_("RRDtool %s"
-           "  Copyright by Tobias Oetiker <tobi@oetiker.ch>\n"
-           "               Compiled %s %s\n\n"
+           "  Copyright by Tobias Oetiker <tobi@oetiker.ch>\n\n"
            "Usage: rrdtool [options] command command_options\n");
 
     const char *help_list =
@@ -281,7 +280,7 @@
         else if (!strcmp(cmd, "pwd"))
             help_cmd = C_PWD;
     }
-    fprintf(stdout, _(help_main), PACKAGE_VERSION, __DATE__, __TIME__);
+    fprintf(stdout, _(help_main), PACKAGE_VERSION);
     fflush(stdout);
     switch (help_cmd) {
     case C_NONE:
