diff -urN odcctools-20090808.orig/ld64/src/Options.cpp odcctools-20090808/ld64/src/Options.cpp
--- odcctools-20090808.orig/ld64/src/Options.cpp	2009-08-08 22:46:09.000000000 +0200
+++ odcctools-20090808/ld64/src/Options.cpp	2014-01-02 14:45:39.219701544 +0100
@@ -28,6 +28,8 @@
 #include <mach/vm_prot.h>
 #include <fcntl.h>
 #include <ctype.h>
+#include <stdio.h>
+#include <unistd.h>
 #include <vector>
 
 #include "config.h"
diff -urN odcctools-20090808.orig/ld64/src/ld.cpp odcctools-20090808/ld64/src/ld.cpp
--- odcctools-20090808.orig/ld64/src/ld.cpp	2014-01-02 14:44:00.236913555 +0100
+++ odcctools-20090808/ld64/src/ld.cpp	2014-01-02 14:44:22.695277206 +0100
@@ -23,6 +23,7 @@
  
 
 #include <stdlib.h>
+#include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
