diff -urN odcctools-20090808.orig/configure odcctools-20090808/configure
--- odcctools-20090808.orig/configure	2014-01-02 14:55:09.428032051 +0100
+++ odcctools-20090808/configure	2014-01-02 14:50:48.747134547 +0100
@@ -5669,7 +5669,7 @@
 _ACEOF
 
 else
-  SSL_LIBS=-lssl
+  SSL_LIBS="-lssl -lcrypto"
 fi
 
 done
diff -urN odcctools-20090808.orig/ld64/Makefile.in odcctools-20090808/ld64/Makefile.in
--- odcctools-20090808.orig/ld64/Makefile.in	2009-07-03 17:33:09.000000000 +0200
+++ odcctools-20090808/ld64/Makefile.in	2014-01-02 14:58:13.614539161 +0100
@@ -47,7 +47,7 @@
 		  -DASPROG="\"$(ASPROG)\"" $(LTO_DEFS)
 
 MYLDFLAGS	= -L$(top_builddir)/libstuff
-MYLIBS		= -lstuff $(SSL_LIBS) $(UUID_LIBS) $(LTO_LIBS)
+MYLIBS		= -lstuff -ldl $(SSL_LIBS) $(UUID_LIBS) $(LTO_LIBS)
 
 MYCOMPILEFLAGS	= $(WARNINGS) $(MYWARNINGS) $(DEFS) $(MYDEFS) \
 	$(CPPFLAGS) $(MYINCLUDES) $(CFLAGS) $(MDYNAMICNOPIC) 
