diff -urN ss5-3.8.9.orig/common/Makefile.in ss5-3.8.9/common/Makefile.in
--- ss5-3.8.9.orig/common/Makefile.in	2011-08-26 14:28:01.000000000 +0200
+++ ss5-3.8.9/common/Makefile.in	2012-07-28 20:13:03.190874781 +0200
@@ -10,7 +10,7 @@
 
 LIBS = -lpthread @LIBS@ @EXTRA_LIBS@ 
 
-CC = gcc
+CC = @CC@
 
 all:	$(COMMONOBJ)
 
diff -urN ss5-3.8.9.orig/Makefile.in ss5-3.8.9/Makefile.in
--- ss5-3.8.9.orig/Makefile.in	2011-08-26 14:28:01.000000000 +0200
+++ ss5-3.8.9/Makefile.in	2012-07-28 20:13:10.309024950 +0200
@@ -10,7 +10,7 @@
 log_path_base   = @log_path_base@
 trc_path_base   = @trc_path_base@
 
-CC = gcc
+CC = @CC@
 
 CFLAGS=@CFLAGS@ -I . @EXTRA_CFLAGS@ 
 
diff -urN ss5-3.8.9.orig/src/Makefile.in ss5-3.8.9/src/Makefile.in
--- ss5-3.8.9.orig/src/Makefile.in	2011-08-26 14:28:01.000000000 +0200
+++ ss5-3.8.9/src/Makefile.in	2012-07-28 20:13:17.991186979 +0200
@@ -24,7 +24,7 @@
 
 LIBS = -lpthread @LIBS@ @EXTRA_LIBS@ 
 
-CC = gcc
+CC = @CC@
 
 all: ss5 $(SRV)
 
