diff -purN ss5-3.8.9.orig/src/SS5Debug.c ss5-3.8.9/src/SS5Debug.c
--- ss5-3.8.9.orig/src/SS5Debug.c	2011-08-26 14:28:01.000000000 +0200
+++ ss5-3.8.9/src/SS5Debug.c	2015-12-02 15:27:12.128755260 +0100
@@ -22,7 +22,7 @@
 #include"SS5Mod_log.h"
 
 
-inline void S5DebugMethodInfo( pid_t pid, struct _SS5ClientInfo ci ) {
+extern inline void S5DebugMethodInfo( pid_t pid, struct _SS5ClientInfo ci ) {
   char logString[256];
 
   snprintf(logString,256 - 1,"[%u] [DEBU] MethodInfo->Ver       %5d.",pid,ci.Ver);          LOGUPDATE()
@@ -32,14 +32,14 @@ inline void S5DebugMethodInfo( pid_t pid
   snprintf(logString,256 - 1,"[%u] [DEBU] MethodInfo->Method    %5d.",pid,ci.Method);       LOGUPDATE()
 }
 
-inline void S5DebugAuthInfo( pid_t pid, struct _SS5ClientInfo ci ) {
+extern inline void S5DebugAuthInfo( pid_t pid, struct _SS5ClientInfo ci ) {
   char logString[256];
 
   snprintf(logString,256 - 1,"[%u] [DEBU] AuthInfo->Username %s.",pid,ci.Username);    LOGUPDATE()
   snprintf(logString,256 - 1,"[%u] [DEBU] AuthInfo->Password xxxxxxxxxxxx.",pid);      LOGUPDATE()
 }
 
-inline void S5DebugRequestInfo( pid_t pid, struct _SS5RequestInfo ri ) {
+extern inline void S5DebugRequestInfo( pid_t pid, struct _SS5RequestInfo ri ) {
   char logString[256];
 
   snprintf(logString,256 - 1,"[%u] [DEBU] RequestInfo->Ver       %5d.",pid,ri.Ver);       LOGUPDATE()
@@ -50,7 +50,7 @@ inline void S5DebugRequestInfo( pid_t pi
   snprintf(logString,256 - 1,"[%u] [DEBU] RequestInfo->DstPort   %5d.",pid,ri.DstPort);   LOGUPDATE()
 }
 
-inline void S5DebugUdpRequestInfo( pid_t pid, struct _SS5RequestInfo ri ) {
+extern inline void S5DebugUdpRequestInfo( pid_t pid, struct _SS5RequestInfo ri ) {
   char logString[256];
 
   snprintf(logString,256 - 1,"[%u] [DEBU] UdpRequestInfo->Rsv       %5d.",pid,ri.udpRsv);        LOGUPDATE()
@@ -60,7 +60,7 @@ inline void S5DebugUdpRequestInfo( pid_t
   snprintf(logString,256 - 1,"[%u] [DEBU] UdpRequestInfo->DstPort   %5d.",pid,ri.udpDstPort);    LOGUPDATE()
 }
 
-inline void S5DebugUpstreamInfo( pid_t pid, struct _SS5RequestInfo ri ) {
+extern inline void S5DebugUpstreamInfo( pid_t pid, struct _SS5RequestInfo ri ) {
   char logString[256];
 
   struct in_addr in;
@@ -70,7 +70,7 @@ inline void S5DebugUpstreamInfo( pid_t p
   snprintf(logString,256 - 1,"[%u] [DEBU] UpstreamInfo->DstPort       %5d.",pid,ri.upDstPort);               LOGUPDATE()
 }
 
-inline void S5DebugFacilities( pid_t pid, struct _SS5Facilities fa ) {
+extern inline void S5DebugFacilities( pid_t pid, struct _SS5Facilities fa ) {
   char logString[256];
 
   snprintf(logString,256 - 1,"[%u] [DEBU] Facilities->Fixup       %s.",pid,fa.Fixup);           LOGUPDATE()
diff -purN ss5-3.8.9.orig/src/SS5Server.c ss5-3.8.9/src/SS5Server.c
--- ss5-3.8.9.orig/src/SS5Server.c	2011-08-26 14:28:01.000000000 +0200
+++ ss5-3.8.9/src/SS5Server.c	2015-12-02 15:26:52.968288601 +0100
@@ -104,7 +104,7 @@ void S5SetDynamic( void )
   NPeers     = 0;
 }
 
-inline UINT S5ChildClose(int exitCode, UINT childSocket, struct _SS5ClientInfo *ci)
+extern inline UINT S5ChildClose(int exitCode, UINT childSocket, struct _SS5ClientInfo *ci)
 {
 #ifdef SS5_USE_GSSAPI
   OM_uint32 tmp;
