diff -urN make-4.1.orig/remake.c make-4.1/remake.c
--- make-4.1.orig/src/remake.c	2014-10-05 18:24:51.000000000 +0200
+++ make-4.1/src/remake.c	2014-11-15 22:08:43.149853184 +0100
@@ -1549,8 +1549,8 @@
   static const char *dirs[] =
     {
 #ifndef _AMIGA
-      "/lib",
-      "/usr/lib",
+      "@LIBDIR@",
+      "@USRLIBDIR@",
 #endif
 #if defined(WINDOWS32) && !defined(LIBDIR)
 /*
@@ -1559,7 +1559,6 @@
  */
 #define LIBDIR "."
 #endif
-      LIBDIR,                   /* Defined by configuration.  */
       0
     };
 
