diff -purN uClibc-ng-1.0.22.orig/utils/Makefile.in uClibc-ng-1.0.22/utils/Makefile.in
--- uClibc-ng-1.0.22.orig/utils/Makefile.in	2017-02-01 21:38:03.000000000 +0100
+++ uClibc-ng-1.0.22/utils/Makefile.in	2017-08-30 13:03:24.226188956 +0200
@@ -45,7 +45,7 @@ CFLAGS-iconv := $(CFLAGS-utils) \
 
 CFLAGS-locale := $(CFLAGS-utils)
 CFLAGS-getconf :=$(CFLAGS-utils) \
-	-DGETCONF_DIR='"$(CURDIR)"'
+	-DGETCONF_DIR='"/dev/null"'
 
 # "make hostutils" flags
 
@@ -73,7 +73,7 @@ BUILD_CFLAGS-iconv.host    := \
 			-I$(top_srcdir)libc/misc/wchar -DL_iconv_main
 
 BUILD_CFLAGS-getconf.host  := \
-				-DGETCONF_DIR='"$(CURDIR)"'
+				-DGETCONF_DIR='"/dev/null"'
 
 # Rules
 
