diff -purN uClibc-ng-1.0.28.orig/libc/sysdeps/linux/common/bits/socket.h uClibc-ng-1.0.28/libc/sysdeps/linux/common/bits/socket.h
--- uClibc-ng-1.0.28.orig/libc/sysdeps/linux/common/bits/socket.h	2018-01-21 12:56:37.000000000 +0100
+++ uClibc-ng-1.0.28/libc/sysdeps/linux/common/bits/socket.h	2018-01-31 23:39:15.743985511 +0100
@@ -81,7 +81,10 @@ typedef __socklen_t socklen_t;
 #define PF_CAIF		37	/* CAIF sockets.  */
 #define PF_ALG		38	/* Algorithm sockets.  */
 #define PF_NFC		39	/* NFC sockets.  */
-#define	PF_MAX		40	/* For now..  */
+#define PF_VSOCK	40	/* vSockets.  */
+#define PF_KCM		41	/* Kernel Connection Multiplexor.  */
+#define PF_QIPCRTR	42	/* Qualcomm IPC Router.  */
+#define 	PF_MAX		43	/* For now..  */
 
 /* Address families.  */
 #define	AF_UNSPEC	PF_UNSPEC
@@ -125,6 +128,9 @@ typedef __socklen_t socklen_t;
 #define AF_CAIF		PF_CAIF
 #define AF_ALG		PF_ALG
 #define AF_NFC		PF_NFC
+#define AF_VSOCK	PF_VSOCK
+#define AF_KCM		PF_KCM
+#define AF_QIPCRTR	PF_QIPCRTR
 #define	AF_MAX		PF_MAX
 
 /* Socket level values.  Others are defined in the appropriate headers.
