diff -purN uClibc-ng-1.0.28.orig/libc/stdio/_fpmaxtostr.c uClibc-ng-1.0.28/libc/stdio/_fpmaxtostr.c
--- uClibc-ng-1.0.28.orig/libc/stdio/_fpmaxtostr.c	2018-01-21 12:56:36.000000000 +0100
+++ uClibc-ng-1.0.28/libc/stdio/_fpmaxtostr.c	2018-03-12 22:29:41.393363608 +0100
@@ -533,9 +533,6 @@ ssize_t _fpmaxtostr(FILE * fp, __fpmax_t
 		if (PRINT_INFO_FLAG_VAL(info,alt)
 			|| (i)
 			|| ((o_mode != 'g')
-#ifdef __UCLIBC_HAS_HEXADECIMAL_FLOATS__
-				&& (o_mode != 'a')
-#endif /* __UCLIBC_HAS_HEXADECIMAL_FLOATS__ */
 				&& (preci > 0))
 			) {
 			ppc[0] = FPO_STR_PREC;
