diff -urN isdnutils-3.25.orig/pppdcapiplugin/Makefile isdnutils-3.25/pppdcapiplugin/Makefile
--- isdnutils-3.25.orig/pppdcapiplugin/Makefile	2012-07-06 08:27:19.000000000 +0200
+++ isdnutils-3.25/pppdcapiplugin/Makefile	2013-11-30 09:44:31.035156307 +0100
@@ -95,3 +95,6 @@
 
 config:
 	@echo nothing to configure
+
+pversion: pversion.c
+	$(HOSTCC) -o $@ $<
diff -urN isdnutils-3.25.orig/pppdcapiplugin/Rules.make isdnutils-3.25/pppdcapiplugin/Rules.make
--- isdnutils-3.25.orig/pppdcapiplugin/Rules.make	2012-07-06 08:27:19.000000000 +0200
+++ isdnutils-3.25/pppdcapiplugin/Rules.make	2013-11-30 09:46:59.396718686 +0100
@@ -13,8 +13,9 @@
 CC	= gcc
 INC     = -I$(TOPDIR) -I$(CAPIINC) -I$(PPPINC)
 DEFS    = -DPPPVER=$(shell $(TOPDIR)/pversion $(PPPVERSION))
-MYCFLAGS= -O2 -Wall -fPIC $(DEFS) $(INC) -L$(CAPILIB)
-LDFLAGS	+= -shared -L$(CAPILIB)
+MYCFLAGS= $(DEFS) $(INC) -fPIC -Wall
+CFLAGS  = -O2
+override LDFLAGS += -shared -fPIC -L$(CAPILIB)
 
 ALL = capiplugin.so userpass.so
 
