diff -purN pppdcapiplugin-3.25+dfsg1.orig/pppdcapiplugin/capiplugin.c pppdcapiplugin-3.25+dfsg1/pppdcapiplugin/capiplugin.c
--- pppdcapiplugin-3.25+dfsg1.orig/pppdcapiplugin/capiplugin.c	2012-07-06 08:27:19.000000000 +0200
+++ pppdcapiplugin-3.25+dfsg1/pppdcapiplugin/capiplugin.c	2017-01-23 21:48:14.041443969 +0100
@@ -133,7 +133,7 @@ static STRINGLIST *clis;
 #define PROTO_ADSLPPPOA		7
 #define PROTO_ADSLPPPOALLC	8
 #define PROTO_ANALOGMODEM	9
-static char *opt_proto = "hdlc";
+static char *opt_proto = NULL;
 static int proto = PROTO_HDLC;
 static int opt_avmadsl = 0;
 static int opt_vpi = -1; /* T-DSL: 1  */
@@ -478,6 +478,9 @@ static void plugin_check_options(void)
 	/*
 	 * protocol
 	 */
+	if (opt_proto == NULL) {
+	   opt_proto = "hdlc";
+	}
 	if (strcasecmp(opt_proto, "hdlc") == 0) {
 	   proto = PROTO_HDLC;
 	} else if (strcasecmp(opt_proto, "x75") == 0) {
