diff -purN u-boot-2015.07.orig/include/configs/rpi-common.h u-boot-2015.07/include/configs/rpi-common.h
--- u-boot-2015.07.orig/include/configs/rpi-common.h	2015-07-14 19:32:21.000000000 +0200
+++ u-boot-2015.07/include/configs/rpi-common.h	2015-11-11 22:44:42.538951995 +0100
@@ -165,17 +165,19 @@
  *   for the FDT/DTB to be up to 1M, which is hopefully plenty.
  */
 #define ENV_MEM_LAYOUT_SETTINGS \
-	"scriptaddr=0x00000000\0" \
+	"fdtcontroladdr=0x00000100\0" \
+	"fdt_addr=0x00000100\0" \
 	"pxefile_addr_r=0x00100000\0" \
 	"kernel_addr_r=0x01000000\0" \
-	"fdt_addr_r=0x02000000\0" \
 	"ramdisk_addr_r=0x02100000\0" \
+	"scriptaddr=0x03000000\0" \
 
 #define BOOT_TARGET_DEVICES(func) \
 	func(MMC, mmc, 0) \
 	func(USB, usb, 0) \
 	func(PXE, pxe, na) \
 	func(DHCP, dhcp, na)
+#define CONFIG_BOOTCOMMAND "fdt addr ${fdt_addr} && fdt get value bootargs /chosen bootargs; run distro_bootcmd"
 #include <config_distro_bootcmd.h>
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
