diff -urN glob.orig/gltests/test-getlogin_r.c glob/gltests/test-getlogin_r.c
--- glob.orig/gltests/test-getlogin_r.c	2014-06-03 15:17:26.362128451 +0200
+++ glob/gltests/test-getlogin_r.c	2014-06-03 11:50:23.634025179 +0200
@@ -28,7 +28,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#if !((defined _WIN32 || defined __WIN32__) && !defined __CYGWIN__)
 #include <pwd.h>
+#endif
 
 #include <sys/stat.h>
 #include <sys/types.h>
