diff -urN isdnutils-3.9.20060704+dfsg.3.orig/capifax/fax.h isdnutils-3.9.20060704+dfsg.3/capifax/fax.h
--- isdnutils-3.9.20060704+dfsg.3.orig/capifax/fax.h	1998-10-23 14:50:54.000000000 +0200
+++ isdnutils-3.9.20060704+dfsg.3/capifax/fax.h	2012-02-28 23:59:39.394766912 +0100
@@ -40,7 +40,7 @@
 	unsigned char len;
 	unsigned short resolution __attribute__ ((packed));
 	unsigned short format __attribute__ ((packed));
-	unsigned char Infos[100] __attribute__ ((packed));
+	unsigned char Infos[100];
 } B3_PROTO_FAXG3;
 
 void SetupB3Config(B3_PROTO_FAXG3 *B3conf, int FAX_Format);
diff -urN isdnutils-3.9.20060704+dfsg.3.orig/capifax/id.h isdnutils-3.9.20060704+dfsg.3/capifax/id.h
--- isdnutils-3.9.20060704+dfsg.3.orig/capifax/id.h	1998-10-23 14:50:56.000000000 +0200
+++ isdnutils-3.9.20060704+dfsg.3/capifax/id.h	2012-02-28 23:58:12.781775236 +0100
@@ -48,8 +48,8 @@
 	_cword resolution  __attribute__ ((packed));
 	_cword format      __attribute__ ((packed));
 	_cword pages       __attribute__ ((packed));
-	_cbyte idlen       __attribute__ ((packed));
-	_cbyte id[50]      __attribute__ ((packed));
+	_cbyte idlen;
+	_cbyte id[50];
 } faxNCPI_t;
 
 extern const char *ConnectionStateString[7];
