diff -urN gobi_loader-0.7.orig/Makefile gobi_loader-0.7/Makefile
--- gobi_loader-0.7.orig/Makefile	2010-06-24 19:13:46.000000000 +0200
+++ gobi_loader-0.7/Makefile	2015-05-04 22:22:19.896119740 +0200
@@ -1,7 +1,11 @@
 VERSION = 0.7
 
+CC = gcc
+CFLAGS =
+LDFLAGS =
+
 gobi_loader: gobi_loader.c
-	gcc -Wall gobi_loader.c -o gobi_loader
+	$(CC) $(CFLAGS) $(LDFLAGS) -Wall gobi_loader.c -o gobi_loader
 
 all: gobi_loader
 
