diff -purN dibbler-1.0.1.orig/Port-linux/lowlevel-options-linux.c dibbler-1.0.1/Port-linux/lowlevel-options-linux.c
--- dibbler-1.0.1.orig/Port-linux/lowlevel-options-linux.c	2015-08-05 22:44:38.000000000 +0200
+++ dibbler-1.0.1/Port-linux/lowlevel-options-linux.c	2016-02-17 22:41:28.911738170 +0100
@@ -232,6 +232,7 @@ int cfg_file_del(const char *file, const
           -2 - unable to open resolv.conf file
  */
 int dns_add(const char * ifname, int ifaceid, const char * addrPlain) {
+#if 0
     FILE * f = NULL;
     unsigned char c;
 
@@ -255,10 +256,12 @@ int dns_add(const char * ifname, int ifa
     }
     fprintf(f,"nameserver %s\n",addrPlain);
     fclose(f);
+#endif
     return LOWLEVEL_NO_ERROR;
 }
 
 int dns_del(const char * ifname, int ifaceid, const char *addrPlain) {
+#if 0
     
 #ifdef MOD_RESOLVCONF
     FILE *f = NULL;
@@ -270,6 +273,9 @@ int dns_del(const char * ifname, int ifa
 #endif
     
     return cfg_file_del(RESOLVCONF_FILE, "nameserver", addrPlain);
+#else
+    return LOWLEVEL_NO_ERROR;
+#endif
 }
 
 int domain_add(const char* ifname, int ifaceid, const char* domain) {
@@ -542,6 +548,7 @@ void delete_radvd_conf(const char* ifnam
 int prefix_add(const char* ifname, int ifindex, const char* prefixPlain, int prefixLength,
 	       unsigned long preferred, unsigned long valid)
 {
+#if 0
     char *argv[5];
     int result;
     char buf[128];
@@ -568,11 +575,15 @@ int prefix_add(const char* ifname, int i
         return LOWLEVEL_NO_ERROR;
     else
         return LOWLEVEL_ERROR_UNSPEC;
+#else
+    return LOWLEVEL_NO_ERROR;
+#endif
 }
 
 int prefix_update(const char* ifname, int ifindex, const char* prefixPlain, int prefixLength,
 		  unsigned long prefered, unsigned long valid)
 {
+#if 0
     char *argv[3];
     int result;
     char buf[128];
@@ -596,10 +607,14 @@ int prefix_update(const char* ifname, in
         return LOWLEVEL_NO_ERROR;
     else
         return LOWLEVEL_ERROR_UNSPEC;
+#else
+    return LOWLEVEL_NO_ERROR;
+#endif
 }
 
 
 int prefix_del(const char* ifname, int ifindex, const char* prefixPlain, int prefixLength) {
+#if 0
 
     int result;
     char *argv[3];
@@ -618,4 +633,7 @@ int prefix_del(const char* ifname, int i
         return LOWLEVEL_NO_ERROR;
     else
         return LOWLEVEL_ERROR_UNSPEC;
+#else
+    return LOWLEVEL_NO_ERROR;
+#endif
 }
