diff -urN dibbler-1.0.0.orig/SrvCfgMgr/SrvCfgMgr.cpp dibbler-1.0.0/SrvCfgMgr/SrvCfgMgr.cpp
--- dibbler-1.0.0.orig/SrvCfgMgr/SrvCfgMgr.cpp	2014-11-08 20:27:57.724165734 +0100
+++ dibbler-1.0.0/SrvCfgMgr/SrvCfgMgr.cpp	2014-11-08 20:31:34.274176106 +0100
@@ -574,9 +574,7 @@
         duid = clientId->getDUID();
     }
 
-    SPtr<TSrvCfgIface> ptrIface;
-    firstIface();
-    while((ptrIface=getIface())&&(ptrIface->getID()!=iface)) ;
+    SPtr<TSrvCfgIface> ptrIface = this->getIfaceByID(iface);
 
     /** @todo: reject-client and accept-only does not work in stateless mode */
     if (this->stateless())
