diff -urN dibbler-1.0.0.orig/IfaceMgr/IfaceMgr.cpp dibbler-1.0.0/IfaceMgr/IfaceMgr.cpp
--- dibbler-1.0.0.orig/IfaceMgr/IfaceMgr.cpp	2014-10-18 20:42:30.000000000 +0200
+++ dibbler-1.0.0/IfaceMgr/IfaceMgr.cpp	2014-11-08 20:28:10.724166362 +0100
@@ -183,7 +183,7 @@
 #else
         sleep(time); // Posix sleep is specified in seconds
 #endif
-        return 0;
+        return -1;
     }
     result = ::select(maxFD, &fds, NULL, NULL, &czas);
 
