diff -purN dibbler-1.0.1.orig/AddrMgr/AddrMgr.cpp dibbler-1.0.1/AddrMgr/AddrMgr.cpp
--- dibbler-1.0.1.orig/AddrMgr/AddrMgr.cpp	2016-06-20 08:29:56.992793916 +0200
+++ dibbler-1.0.1/AddrMgr/AddrMgr.cpp	2016-06-20 08:55:17.485153886 +0200
@@ -839,13 +839,14 @@ SPtr<TAddrClient> TAddrMgr::parseAddrCli
                 ifindex = atoi(x+7);
                 // Log(Debug) << "Parsed AddrPD::iface=" << iface << LogEnd;
             }
-            if (strstr(buf,"unicast")) {
-                x = strstr(buf,"=")+2;
-                if (x)
-                    x = strstr(x," ")-1;
-                if (x)
-                    *x = 0; // remove trailing xml tag
-                unicast = new TIPv6Addr(strstr(buf,"=")+2, true);
+            if ((x=strstr(buf,"unicast"))) {
+                char *end = strstr(x+9, "\"");
+                if (end) {
+                    string uni(x+9, end);
+                    if (uni.size()) {
+                        unicast = new TIPv6Addr(uni.c_str(), true);
+                    }
+                }
             }
             if ((x=strstr(buf,"ifacename"))) {
                 char* end = strstr(x + 11, "\"");
