--- iwleeprom/Makefile	2014-10-17 21:19:53.902516338 +0200
+++ iwleeprom.patched/Makefile	2014-10-18 10:10:31.075108657 +0200
@@ -11,19 +11,19 @@
 		mkdir $(OBJDIR)
 
 iwleeprom: $(OBJDIR)/iwlio.o $(OBJDIR)/ath5kio.o $(OBJDIR)/ath9kio.o $(OBJDIR)/iwleeprom.o
-		gcc -Wall $(CFLAGS) -o iwleeprom $(OBJDIR)/iwleeprom.o $(OBJDIR)/iwlio.o $(OBJDIR)/ath5kio.o $(OBJDIR)/ath9kio.o
+		$(CC) -Wall $(CFLAGS) -o iwleeprom $(OBJDIR)/iwleeprom.o $(OBJDIR)/iwlio.o $(OBJDIR)/ath5kio.o $(OBJDIR)/ath9kio.o
 
 $(OBJDIR)/iwleeprom.o: iwleeprom.h iwlio.h ath5kio.h ath9kio.h iwleeprom.c
-		gcc -Wall $(CFLAGS) -c -o $(OBJDIR)/iwleeprom.o iwleeprom.c
+		$(CC) -Wall $(CFLAGS) -c -o $(OBJDIR)/iwleeprom.o iwleeprom.c
 
 $(OBJDIR)/iwlio.o: iwleeprom.h iwlio.h iwlio.c
-		gcc -Wall $(CFLAGS) -c -o $(OBJDIR)/iwlio.o iwlio.c
+		$(CC) -Wall $(CFLAGS) -c -o $(OBJDIR)/iwlio.o iwlio.c
 
 $(OBJDIR)/ath5kio.o: iwleeprom.h ath5kio.h ath5kio.c
-		gcc -Wall $(CFLAGS) -c -o $(OBJDIR)/ath5kio.o ath5kio.c
+		$(CC) -Wall $(CFLAGS) -c -o $(OBJDIR)/ath5kio.o ath5kio.c
 
 $(OBJDIR)/ath9kio.o: iwleeprom.h ath9kio.h ath9kio.c
-		gcc -Wall $(CFLAGS) -c -o $(OBJDIR)/ath9kio.o ath9kio.c
+		$(CC) -Wall $(CFLAGS) -c -o $(OBJDIR)/ath9kio.o ath9kio.c
 
 iwleeprom.8.gz: iwleeprom.8
 		gzip -c iwleeprom.8 > iwleeprom.8.gz
