diff -urN gettext-0.16.1.orig/gettext-runtime/intl/Makefile.in gettext-0.16.1/gettext-runtime/intl/Makefile.in
--- gettext-0.16.1.orig/gettext-runtime/intl/Makefile.in	2006-11-27 18:02:00.000000000 +0100
+++ gettext-0.16.1/gettext-runtime/intl/Makefile.in	2012-07-27 23:38:44.387135164 +0200
@@ -320,7 +320,7 @@
 	  $(mkdir_p) $(DESTDIR)$(libdir) $(DESTDIR)$(includedir); \
 	  $(INSTALL_DATA) libintl.h $(DESTDIR)$(includedir)/libintl.h; \
 	  $(LIBTOOL) --mode=install \
-	    $(INSTALL_DATA) libintl.$la $(DESTDIR)$(libdir)/libintl.$la; \
+	    $(INSTALL) libintl.$la $(DESTDIR)$(libdir)/libintl.$la; \
 	  if test "@RELOCATABLE@" = yes; then \
 	    dependencies=`sed -n -e 's,^dependency_libs=\(.*\),\1,p' < $(DESTDIR)$(libdir)/libintl.la | sed -e "s,^',," -e "s,'\$$,,"`; \
 	    if test -n "$$dependencies"; then \
@@ -335,9 +335,9 @@
 	   && test @GLIBC2@ != no; then \
 	  $(mkdir_p) $(DESTDIR)$(libdir); \
 	  $(LIBTOOL) --mode=install \
-	    $(INSTALL_DATA) libgnuintl.$la $(DESTDIR)$(libdir)/libgnuintl.$la; \
+	    $(INSTALL) libgnuintl.$la $(DESTDIR)$(libdir)/libgnuintl.$la; \
 	  rm -f $(DESTDIR)$(libdir)/preloadable_libintl.so; \
-	  $(INSTALL_DATA) $(DESTDIR)$(libdir)/libgnuintl.so $(DESTDIR)$(libdir)/preloadable_libintl.so; \
+	  $(INSTALL) $(DESTDIR)$(libdir)/libgnuintl.so $(DESTDIR)$(libdir)/preloadable_libintl.so; \
 	  $(LIBTOOL) --mode=uninstall \
 	    rm -f $(DESTDIR)$(libdir)/libgnuintl.$la; \
 	else \
