diff -urN wireless_tools.29.orig/Makefile wireless_tools.29/Makefile
--- wireless_tools.29.orig/Makefile	2007-09-18 01:56:46.000000000 +0200
+++ wireless_tools.29/Makefile	2013-11-30 11:39:33.746364213 +0100
@@ -145,7 +145,7 @@
 
 # Compilation of the dynamic library
 $(DYNAMIC): $(OBJS:.o=.so)
-	$(CC) -shared -o $@ -Wl,-soname,$@ $(STRIPFLAGS) $(LIBS) -lc $^
+	$(CC) $(LDFLAGS) -shared -o $@ -Wl,-soname,$@ $(STRIPFLAGS) $(LIBS) -lc $^
 
 # Compilation of the static library
 $(STATIC): $(OBJS:.o=.so)
