diff -urN src.orig/src/liblzma/api/lzma/base.h src/src/liblzma/api/lzma/base.h
--- src.orig/src/liblzma/api/lzma/base.h	2011-05-21 14:12:31.000000000 +0200
+++ src/src/liblzma/api/lzma/base.h	2012-07-06 14:18:44.000000000 +0200
@@ -215,7 +215,7 @@
 		 * applications a little.
 		 */
 
-	LZMA_PROG_ERROR         = 11,
+	LZMA_PROG_ERROR         = 11
 		/**<
 		 * \brief       Programming error
 		 *
