diff -urN ucarp-1.5.2.orig/src/ucarp.c ucarp-1.5.2/src/ucarp.c
--- ucarp-1.5.2.orig/src/ucarp.c	2009-05-27 23:42:44.000000000 +0200
+++ ucarp-1.5.2/src/ucarp.c	2012-08-08 14:16:23.007706420 +0200
@@ -21,7 +21,7 @@
 
 static void usage(void)
 {
-    puts("\n" PACKAGE_STRING " - " __DATE__ "\n");
+    puts("\n" PACKAGE_STRING "\n");
     fputs(_(
         "--interface=<if> (-i <if>): bind interface <if>\n"
         "--srcip=<ip> (-s <ip>): source (real) IP address of that host\n"
