diff -urN ucarp-1.5.2.orig/configure.ac ucarp-1.5.2/configure.ac
--- ucarp-1.5.2.orig/configure.ac	2010-01-31 22:55:49.000000000 +0100
+++ ucarp-1.5.2/configure.ac	2012-07-29 10:19:28.639223762 +0200
@@ -200,63 +200,12 @@
 fi
 
 AC_MSG_CHECKING(whether snprintf is C99 conformant)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-#ifdef STDC_HEADERS
-# include <stdlib.h>
-# include <stddef.h>
-#else
-# if HAVE_STDLIB_H
-#  include <stdlib.h>
-# endif
-#endif
-#ifdef HAVE_UNISTD_H
-# include <unistd.h>
-#endif
-
-int main(void)
-{
-    char buf[4];
-    
-    (void) fprintf(fopen("conftestval", "w"), "%d\n",
-        (int) snprintf(buf, sizeof buf, "12345678"));
-    return 0;
-}
-]])],[CONF_SNPRINTF_TYPE=`cat conftestval`
-],[],[])
 AC_MSG_RESULT(done)
-if test "x$CONF_SNPRINTF_TYPE" = "x" ; then
-  AC_MSG_WARN(your operating system doesn't implement snprintf)
-else
-  AC_DEFINE_UNQUOTED(CONF_SNPRINTF_TYPE, $CONF_SNPRINTF_TYPE,
+  AC_DEFINE_UNQUOTED(CONF_SNPRINTF_TYPE, 8,
                      [return value of an overflowed snprintf])
-fi
 
 AC_MSG_CHECKING(whether you already have a standard SHA1 implementation)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-#include <string.h>
-#include <sys/types.h>
-#include <sha1.h>
-
-int main(void)
-{
-    SHA1_CTX ctx;
-    char b[41];
-    
-    SHA1Init(&ctx);
-    SHA1Update(&ctx, (const unsigned char *) "test", 4U);
-    SHA1End(&ctx, b);
-    b[40] = 0;
-    
-    return strcasecmp(b, "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3");
-}
-]])],[
-AC_MSG_RESULT(yes)
-AC_DEFINE(USE_SYSTEM_CRYPT_SHA1,,[Define if you already have standard
-SHA1 functions])
-],[AC_MSG_RESULT(no)
-],[])
+AC_MSG_RESULT(no)
 
 AC_MSG_CHECKING([whether syslog names are available])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
