diff -urN kamailio-4.0.1.orig/Makefile kamailio-4.0.1/Makefile
--- kamailio-4.0.1.orig/Makefile	2013-04-25 16:56:29.000000000 +0200
+++ kamailio-4.0.1/Makefile	2013-05-28 22:59:47.239973752 +0200
@@ -328,7 +328,7 @@
 space=$(nullstring) $(nullstring)
 
 modules_search_path=$(subst $(space),:,$(strip\
-						$(foreach m,$(modules_dirs),$($(m)_target))))
+						$(foreach m,$(modules_dirs),$(patsubst $(basedir)%,%,$($(m)_target)))))
 		#				$(addprefix $(modules_target),$(modules_dirs))))
 
 # special depends for main.o
diff -urN kamailio-4.0.1.orig/Makefile.defs kamailio-4.0.1/Makefile.defs
--- kamailio-4.0.1.orig/Makefile.defs	2013-04-25 16:56:29.000000000 +0200
+++ kamailio-4.0.1/Makefile.defs	2013-05-28 23:17:45.792161757 +0200
@@ -440,7 +440,7 @@
 share_dir = share/$(MAIN_NAME)/
 # lib/$(MAIN_NAME)/modules , lib/$(MAIN_NAME)/modules-s, lib/$(MAIN_NAME)/modules-k
 modules_dir = $(LIBDIR)/$(MAIN_NAME)/
-lib_dir = $(LIBDIR)/$(MAIN_NAME)/
+lib_dir = $(LIBDIR)/
 
 doc_dir = doc/$(MAIN_NAME)/
 man_dir = man/
@@ -505,7 +505,7 @@
 BASEDIR ?= $(DESTDIR)
 basedir = $(BASEDIR)
 # install prefixes for various stuff
-cfg_prefix = $(basedir)$(prefix)
+cfg_prefix = $(basedir)
 bin_prefix = $(basedir)$(prefix)
 modules_prefix = $(basedir)$(prefix)
 lib_prefix = $(basedir)$(prefix)
@@ -517,7 +517,7 @@
 
 
 # target dirs for various stuff
-cfg_target = $(prefix)/$(cfg_dir)
+cfg_target = /$(cfg_dir)
 bin_target = $(prefix)/$(bin_dir)
 #modules_target = $(prefix)/$(modules_dir)
 lib_target = $(prefix)/$(lib_dir)
