diff -urN kamailio-4.0.1.orig/utils/sercmd/Makefile kamailio-4.0.1/utils/sercmd/Makefile
--- kamailio-4.0.1.orig/utils/sercmd/Makefile	2013-04-25 16:56:29.000000000 +0200
+++ kamailio-4.0.1/utils/sercmd/Makefile	2013-05-28 22:02:01.518811822 +0200
@@ -15,7 +15,7 @@
 readline_locations= /usr/include/readline/readline.h \
 					 $(LOCALBASE)/include/readline/readline.h
 
-use_readline ?=
+use_readline := 0
 ifneq (,$(MAKECMDGOALS))
 ifeq (,$(filter-out $(nodep_targets),$(MAKECMDGOALS)))
 #set it to empty, we don't need to detect/use it for clean, doc a.s.o
