diff -urN kamailio-4.0.4.orig/core_cmd.c kamailio-4.0.4/core_cmd.c
--- kamailio-4.0.4.orig/core_cmd.c	2013-10-02 17:01:05.000000000 +0200
+++ kamailio-4.0.4/core_cmd.c	2013-11-20 21:10:17.763624175 +0100
@@ -369,8 +369,6 @@
 	if (rpc->add(c, "{", &s) < 0) return;
 	rpc->struct_printf(s, "version", "%s %s", ver_name, ver_version);
 	rpc->struct_add(s, "s", "id", ver_id);
-	rpc->struct_add(s, "s", "compiler", ver_compiler);
-	rpc->struct_add(s, "s", "compiled", ver_compiled_time);
 	rpc->struct_add(s, "s", "flags", ver_flags);
 }
 
diff -urN kamailio-4.0.4.orig/main.c kamailio-4.0.4/main.c
--- kamailio-4.0.4.orig/main.c	2013-10-02 17:01:05.000000000 +0200
+++ kamailio-4.0.4/main.c	2013-11-20 21:10:17.766624259 +0100
@@ -306,8 +306,6 @@
 	printf("  TCP poll methods: %s\n", poll_support);
 #endif
 	printf("  Source code revision ID: %s\n", ver_id);
-	printf("  Compiled with: %s\n", ver_compiler);
-	printf("  Compiled on: %s\n", ver_compiled_time);
 	printf("Thank you for flying %s!\n", NAME);
 }
 
@@ -1970,8 +1968,6 @@
					printf("flags: %s\n", ver_flags );
					print_ct_constants();
 					printf("id: %s\n", ver_id);
-					printf("compiled on %s with %s\n",
-							ver_compiled_time, ver_compiler );
 
 					exit(0);
 					break;
diff -urN kamailio-4.0.4.orig/modules/kex/mi_core.c kamailio-4.0.4/modules/kex/mi_core.c
--- kamailio-4.0.4.orig/modules/kex/mi_core.c	2013-10-02 17:01:04.000000000 +0200
+++ kamailio-4.0.4/modules/kex/mi_core.c	2013-11-20 21:11:09.938030429 +0100
@@ -51,7 +51,7 @@
 #include "../../cfg/cfg.h"
 #include "../../cfg/cfg_ctx.h"
 
-#define BUILD_STR __FILE__ " compiled on "__TIME__ " " __DATE__ " with " COMPILER "\n"
+#define BUILD_STR __FILE__ " compiled with " COMPILER "\n"
 #define BUILD_STR_LEN (sizeof(BUILD_STR)-1)
 
 #ifndef SVNREVISION
diff -urN kamailio-4.0.4.orig/utils/sercmd/sercmd.c kamailio-4.0.4/utils/sercmd/sercmd.c
--- kamailio-4.0.4.orig/utils/sercmd/sercmd.c	2013-10-02 17:01:04.000000000 +0200
+++ kamailio-4.0.4/utils/sercmd/sercmd.c	2013-11-20 21:10:25.529839463 +0100
@@ -86,7 +86,6 @@
 
 static char id[]="$Id$";
 static char version[]= NAME " " VERSION;
-static char compiled[]= __TIME__ " " __DATE__;
 static char help_msg[]="\
 Usage: " NAME " [options][-s address] [ cmd ]\n\
 Options:\n\
@@ -1769,7 +1768,6 @@
 {
 	printf("%s\n", version);
 	printf("%s\n", id);
-	printf("%s compiled on %s \n", __FILE__, compiled);
 #ifdef USE_READLINE
 	printf("interactive mode command completion support\n");
 #endif
@@ -2174,8 +2172,6 @@
 			case 'V':
 				printf("version: %s\n", version);
 				printf("%s\n", id);
-				printf("%s compiled on %s \n", __FILE__,
-						compiled);
 				exit(0);
 				break;
 			case 'h':
diff -urN kamailio-4.0.4.orig/ver.c kamailio-4.0.4/ver.c
--- kamailio-4.0.4.orig/ver.c	2013-10-02 17:01:04.000000000 +0200
+++ kamailio-4.0.4/ver.c	2013-11-20 21:10:18.658649095 +0100
@@ -35,8 +35,6 @@
 const char ver_arch[] = ARCH;
 const char ver_os[] = OS_QUOTED;
 const char ver_id[] = REPO_HASH " " REPO_STATE;
-const char ver_compiled_time[] =  __TIME__ " " __DATE__ ;
-const char ver_compiler[] = COMPILER;
 
 const char ver_flags[] = SER_COMPILE_FLAGS;
 
diff -urN kamailio-4.0.4.orig/ver.h kamailio-4.0.4/ver.h
--- kamailio-4.0.4.orig/ver.h	2013-10-02 17:01:04.000000000 +0200
+++ kamailio-4.0.4/ver.h	2013-11-20 21:10:25.485838249 +0100
@@ -34,8 +34,6 @@
 extern const char ver_arch[];
 extern const char ver_os[];
 extern const char ver_id[];
-extern const char ver_compiled_time[];
-extern const char ver_compiler[];
 extern const char ver_flags[];
 extern const char repo_ver[];
 extern const char repo_hash[];
