diff -urN traceroute-2.0.18/Make.rules traceroute-2.0.18.patched/Make.rules
--- traceroute-2.0.18/Make.rules	2010-09-09 16:20:46.000000000 +0200
+++ traceroute-2.0.18.patched/Make.rules	2012-04-01 11:45:27.000000000 +0200
@@ -17,7 +17,7 @@
 endif
 
 
-MAKE = make --no-print-directory -r
+MAKEFLAGS += --no-print-directory -r
 
 
 #  Use env=yes on cmdline to inherit environment values
@@ -45,7 +45,7 @@
 $(call set, INSTALL, cp)
 $(call set, INDENT, true)
 
-gcc = $(findstring gcc,$(CC))
+#gcc = $(findstring gcc,$(CC))
 $(call set, CFLAGS, $(if $(gcc), -O2 -Wall, -O))
 $(call set, CPPFLAGS, )
 $(call set, LDFLAGS, -s)
