diff -urN gogoc-1_2-RELEASE.orig/gogoc-pal/platform/common/src/pal_version.c gogoc-1_2-RELEASE/gogoc-pal/platform/common/src/pal_version.c
--- gogoc-1_2-RELEASE.orig/gogoc-pal/platform/common/src/pal_version.c	2009-11-20 17:38:49.000000000 +0100
+++ gogoc-1_2-RELEASE/gogoc-pal/platform/common/src/pal_version.c	2012-08-08 14:27:05.501137851 +0200
@@ -13,7 +13,7 @@
 */
 #include "pal_version.h"
 
-#define PAL_VER_STRING "gogoCLIENT PAL for " PLATFORM " built on " __DATE__
+#define PAL_VER_STRING "gogoCLIENT PAL for " PLATFORM
 
 /*
   Returns the pal version.
diff -urN gogoc-1_2-RELEASE.orig/gogoc-tsp/include/version.h gogoc-1_2-RELEASE/gogoc-tsp/include/version.h
--- gogoc-1_2-RELEASE.orig/gogoc-tsp/include/version.h	2010-04-20 19:16:35.000000000 +0200
+++ gogoc-1_2-RELEASE/gogoc-tsp/include/version.h	2012-08-08 14:26:26.014545696 +0200
@@ -33,8 +33,8 @@
 #define TSP_CLIENT_VERSION "1.2-RELEASE"
 
 // Identification string.
-#define IDENTIFICATION "gogoCLIENT v" TSP_CLIENT_VERSION " build " __DATE__ \
-        "-" __TIME__ " " TSP_CLIENT_OPT_SOFT " " TSP_CLIENT_OPT_ARCH
+#define IDENTIFICATION "gogoCLIENT v" TSP_CLIENT_VERSION \
+        " " TSP_CLIENT_OPT_SOFT " " TSP_CLIENT_OPT_ARCH
 
 
 // defined in tsp_client.c
