diff -urN gogoc-1_2-RELEASE.orig/gogoc-tsp/platform/unix-common/Makefile gogoc-1_2-RELEASE/gogoc-tsp/platform/unix-common/Makefile
--- gogoc-1_2-RELEASE.orig/gogoc-tsp/platform/unix-common/Makefile	2009-11-20 17:53:30.000000000 +0100
+++ gogoc-1_2-RELEASE/gogoc-tsp/platform/unix-common/Makefile	2014-12-10 21:11:27.583068299 +0100
@@ -27,7 +27,7 @@
 
 $(OS_UNAME):
 	@echo > $(OS_UNAME)
-	@echo \#define OS_UNAME_INFO \"`uname -a`\" >> $(OS_UNAME)
+	@echo \#undef OS_UNAME_INFO >> $(OS_UNAME)
 	@echo >> $(OS_UNAME)
 
 $(OBJS_DIR)/unix-main.o:unix-main.c $(OS_UNAME)
