From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 20 Jul 2020 09:35:46 +0100
Subject: SQUASH: leds: actpwr - delete unused variable

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 drivers/leds/trigger/ledtrig-actpwr.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/leds/trigger/ledtrig-actpwr.c b/drivers/leds/trigger/ledtrig-actpwr.c
index 4e89fb314e2c..1a52107ceb03 100644
--- a/drivers/leds/trigger/ledtrig-actpwr.c
+++ b/drivers/leds/trigger/ledtrig-actpwr.c
@@ -94,7 +94,6 @@ static void actpwr_trig_cycle(struct timer_list *t)
 {
 	struct actpwr_trig_data *trig  = &actpwr_data;
 	struct actpwr_vled *active;
-	enum led_brightness value;
 
 	active = &trig->virt_leds[trig->next_active];
 	trig->active = active;
