diff -purN bind-9.11.0-P1.orig/bin/named/include/named/globals.h bind-9.11.0-P1/bin/named/include/named/globals.h
--- bind-9.11.0-P1.orig/bin/named/include/named/globals.h	2016-10-21 07:13:38.000000000 +0200
+++ bind-9.11.0-P1/bin/named/include/named/globals.h	2017-03-04 16:38:38.162846480 +0100
@@ -65,8 +65,6 @@ EXTERN const char *		ns_g_version		INIT(
 EXTERN const char *		ns_g_product		INIT(PRODUCT);
 EXTERN const char *		ns_g_description	INIT(DESCRIPTION);
 EXTERN const char *		ns_g_srcid		INIT(SRCID);
-EXTERN const char *		ns_g_configargs		INIT(CONFIGARGS);
-EXTERN const char *		ns_g_builder		INIT(BUILDER);
 EXTERN in_port_t		ns_g_port		INIT(0);
 EXTERN isc_dscp_t		ns_g_dscp		INIT(-1);
 EXTERN in_port_t		lwresd_g_listenport	INIT(0);
diff -purN bind-9.11.0-P1.orig/bin/named/main.c bind-9.11.0-P1/bin/named/main.c
--- bind-9.11.0-P1.orig/bin/named/main.c	2016-10-21 07:13:38.000000000 +0200
+++ bind-9.11.0-P1/bin/named/main.c	2017-03-04 16:39:37.728443295 +0100
@@ -666,8 +666,6 @@ parse_command_line(int argc, char *argv[
 			       (*ns_g_description != '\0') ? " " : "",
 			       ns_g_description, ns_g_srcid);
 			printf("running on %s\n", ns_os_uname());
-			printf("built by %s with %s\n",
-			       ns_g_builder, ns_g_configargs);
 #ifdef __clang__
 			printf("compiled by CLANG %s\n", __VERSION__);
 #else
@@ -1052,9 +1050,6 @@ setup(void) {
 		      ISC_LOG_NOTICE, "running on %s", ns_os_uname());
 
 	isc_log_write(ns_g_lctx, NS_LOGCATEGORY_GENERAL, NS_LOGMODULE_MAIN,
-		      ISC_LOG_NOTICE, "built with %s", ns_g_configargs);
-
-	isc_log_write(ns_g_lctx, NS_LOGCATEGORY_GENERAL, NS_LOGMODULE_MAIN,
 		      ISC_LOG_NOTICE, "running as: %s%s",
 		      program_name, saved_command_line);
 
