diff -purN u-boot-2016.07.orig/include/configs/rpi.h u-boot-2016.07/include/configs/rpi.h
--- u-boot-2016.07.orig/include/configs/rpi.h	2016-07-11 21:01:01.000000000 +0200
+++ u-boot-2016.07/include/configs/rpi.h	2016-08-20 11:43:35.586879691 +0200
@@ -195,7 +195,8 @@
 #define ENV_MEM_LAYOUT_SETTINGS \
 	"fdt_high=ffffffff\0" \
 	"initrd_high=ffffffff\0" \
-	"fdt_addr_r=0x00000100\0" \
+	"fdtcontroladdr=0x00000100\0" \
+	"fdt_addr=0x00000100\0" \
 	"pxefile_addr_r=0x00100000\0" \
 	"kernel_addr_r=0x01000000\0" \
 	"scriptaddr=0x02000000\0" \
@@ -206,6 +207,7 @@
 	func(USB, usb, 0) \
 	func(PXE, pxe, na) \
 	func(DHCP, dhcp, na)
+#define CONFIG_BOOTCOMMAND "fdt addr ${fdt_addr} && fdt get value bootargs /chosen bootargs; run distro_bootcmd"
 #include <config_distro_bootcmd.h>
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
