From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 6 May 2020 17:02:26 +0100
Subject: ARM: dts: Enable firmware-clocks on all Pis

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm270x.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/bcm270x.dtsi b/arch/arm/boot/dts/bcm270x.dtsi
index bf90fd585364..d2f5ea1f31f8 100644
--- a/arch/arm/boot/dts/bcm270x.dtsi
+++ b/arch/arm/boot/dts/bcm270x.dtsi
@@ -7,6 +7,12 @@
 		/delete-property/ stdout-path;
 	};
 
+	firmware_clocks: firmware-clocks {
+		compatible = "raspberrypi,firmware-clocks";
+		raspberrypi,firmware = <&firmware>;
+		#clock-cells = <1>;
+	};
+
 	soc: soc {
 
 		watchdog: watchdog@7e100000 {
