diff --git a/drivers/usb/host/dwc_common_port/dwc_common_linux.c b/drivers/usb/host/dwc_common_port/dwc_common_linux.c
--- a/drivers/usb/host/dwc_common_port/dwc_common_linux.c
+++ b/drivers/usb/host/dwc_common_port/dwc_common_linux.c
@@ -1374,6 +1374,7 @@
 EXPORT_SYMBOL(DWC_THREAD_SHOULD_STOP);
 EXPORT_SYMBOL(DWC_TASK_ALLOC);
 EXPORT_SYMBOL(DWC_TASK_FREE);
+EXPORT_SYMBOL(DWC_TASK_HI_SCHEDULE);
 EXPORT_SYMBOL(DWC_TASK_SCHEDULE);
 EXPORT_SYMBOL(DWC_WORKQ_WAIT_WORK_DONE);
 EXPORT_SYMBOL(DWC_WORKQ_ALLOC);
