diff -purN libcap-2.25.orig/Make.Rules libcap-2.25/Make.Rules
--- libcap-2.25.orig/Make.Rules	2016-06-16 16:05:06.556940839 +0200
+++ libcap-2.25/Make.Rules	2016-06-16 16:08:19.562171492 +0200
@@ -61,7 +61,7 @@ WARNINGS=-Wall -Wwrite-strings \
         -Wnested-externs -Winline -Wshadow
 LD=$(CC) -Wl,-x -shared
 LDFLAGS ?= #-g
-BUILD_GPERF := $(shell which gperf >/dev/null 2>/dev/null && echo yes)
+BUILD_GPERF := no
 
 SYSTEM_HEADERS = /usr/include
 INCS=$(topdir)/libcap/include/sys/capability.h
