Index: capifaxrcvd.c
===================================================================
RCS file: /home/jw5/local/cvs/isdn/isdn4k-utils/capifax/capifaxrcvd.c,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 capifaxrcvd.c
--- capifaxrcvd.c	14 Aug 2003 08:28:26 -0000	1.1.1.1
+++ capifaxrcvd.c	26 Oct 2003 20:27:41 -0000
@@ -154,7 +154,7 @@
 	}
 	if (Slot == INVALID_CONNECTION_ID) {
 		Slot = Connection;
-		sprintf(RcvName, "rc-%s-%08lx", CallingPartyNumber, time(NULL));
+		sprintf(RcvName, "rc-%16s-%08lx", CallingPartyNumber, time(NULL));
 		f = fopen(RcvName, "wb");
 		if (f != NULL) {
 			FileReceive = TRUE;
@@ -208,6 +208,7 @@
 		int i;
 		int l;
 		char newname[1024];
+		char fullname[1024];
 
 		sprintf(newname, "f%c%03d%08lx", (rres)?'f':'n', rpages,
 			time(NULL));
@@ -239,11 +240,13 @@
 			}
 		}
 		newname[l] = '\0';
+		strcpy(fullname, newname);
+		newname[22] = '\0';
 		rename(RcvName, newname);
 		chmod(newname, 0640);
 		syslog(LOG_INFO, "Received a FAX\n");
 		if (notifyCmd) {
-			sprintf(RcvName, "%s %s", notifyCmd, newname);
+			sprintf(RcvName, "%s %s %s", notifyCmd, fullname, newname);
 			system(RcvName);
 		}
 	} else
