diff -uNr usb-modeswitch-2.5.2-a/usb_modeswitch.c usb-modeswitch-2.5.2-b/usb_modeswitch.c
--- usb-modeswitch-2.5.2-a/usb_modeswitch.c	2017-12-31 11:33:30.000000000 +0100
+++ usb-modeswitch-2.5.2-b/usb_modeswitch.c	2018-02-01 16:10:15.000000000 +0100
@@ -1752,6 +1752,8 @@
 			if (product == descriptor.idProduct) {
 				if (verbose)
 					fprintf(output,"   product ID matched\n");
+				if (mode == SEARCH_TARGET)
+					TargetProduct = product;

 				if (targetClass != 0) {
 					/* TargetClass is set, check class of first interface */
