diff -urN cups-1.3.11.orig/configure cups-1.3.11/configure
--- cups-1.3.11.orig/configure	2009-07-03 00:11:22.000000000 +0200
+++ cups-1.3.11/configure	2013-11-17 17:18:06.933313835 +0100
@@ -10191,7 +10191,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest.$ac_objext; then
-  OPTIM="$OPTIM -fstack-protector"
+#  OPTIM="$OPTIM -fstack-protector"
 		{ echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 else
