diff -urN pptpd-1.4.0.orig/Makefile.in pptpd-1.4.0/Makefile.in
--- pptpd-1.4.0.orig/Makefile.in	2013-10-23 09:02:37.000000000 +0200
+++ pptpd-1.4.0/Makefile.in	2015-03-29 19:49:49.051682304 +0200
@@ -153,7 +153,8 @@
 AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = -O2 -fno-builtin -Wall -DSBINDIR='"$(sbindir)"'
+COPTS = -O2 -fno-builtin -Wall
+CFLAGS = $(COPTS) -DSBINDIR='"$(sbindir)"'
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CYGPATH_W = @CYGPATH_W@
diff -urN pptpd-1.4.0.orig/plugins/Makefile pptpd-1.4.0/plugins/Makefile
--- pptpd-1.4.0.orig/plugins/Makefile	2015-03-29 19:49:28.373215845 +0200
+++ pptpd-1.4.0/plugins/Makefile	2015-03-29 20:25:38.802253031 +0200
@@ -1,7 +1,7 @@
 CC	= gcc
 COPTS	= -O2 -g
 CFLAGS  += $(COPTS) -I.. -I../../include -fPIC
-LDFLAGS	+= -shared
+override LDFLAGS += -shared
 LDADD   = -lutil
 INSTALL	= install -o root
 prefix  = /usr/local
