diff -urN e3-2.82.orig/Makefile e3-2.82/Makefile
--- e3-2.82.orig/Makefile	2016-02-21 01:57:38.000000000 +0100
+++ e3-2.82/Makefile	2017-09-07 00:44:39.141478986 +0200
@@ -18,6 +18,7 @@
 #NASM_OR_YASM=yasm
 
 EXMODE=SED
+LD=ld
 
 #----------------------do not edit below line-------------------------------
 
@@ -40,15 +41,14 @@
 #---------------- 32 bit LINUX (and perhaps BSD) --
 
 32:	$(ASOURCES) Makefile
-ifeq	($(OS),LINUX)
-	nasm -O2 -f bin -l e3.lst -o e3 e3.asm -DCRIPLED_ELF=1 -D$(OS) -D$(EXMODE) 
-	chmod +x e3
-else
-	echo $(ASVER)
+#ifeq	($(OS),LINUX)
+#	nasm -O2 -f bin -l e3.lst -o e3 e3.asm -DCRIPLED_ELF=1 -D$(OS) -D$(EXMODE) 
+#	chmod +x e3
+#else
+#	echo $(ASVER)
 	nasm -O2 $(AFLAGS) -o e3.o e3.asm -l e3.lst -D$(OS) -D$(EXMODE)
-	ld -s -o e3 e3.o
-	strip --remove-section .comment e3
-endif
+	$(LD) $(LDFLAGS) -m elf_i386 -o e3 e3.o
+#endif
 	ln -sf e3 e3ws
 	ln -sf e3 e3em
 	ln -sf e3 e3pi
@@ -60,10 +60,7 @@
 64:	$(ASOURCES) Makefile
 	cat e3.asm | sed -f e3_32to64.sed  >e3.tmp
 	$(ASSEMBLER64)
-	ld -b elf64-x86-64 --oformat elf64-x86-64 -o e3 e3.o
-ifneq	($(DEBUG),true)
-	strip e3
-endif
+	$(LD) $(LDFLAGS) -m elf_x86_64 -o e3 e3.o
 	ln -sf e3 e3ws
 	ln -sf e3 e3em
 	ln -sf e3 e3pi
