diff -purN linux-5.3.orig/drivers/acpi/osl.c linux-5.3/drivers/acpi/osl.c
--- linux-5.3.orig/drivers/acpi/osl.c	2019-09-15 23:19:32.000000000 +0200
+++ linux-5.3/drivers/acpi/osl.c	2019-09-17 10:07:55.089941814 +0200
@@ -1438,6 +1438,8 @@ int acpi_check_resource_conflict(const s
 	u8 warn = 0;
 	int clash = 0;
 
+	if (acpi_disabled)
+		return 0;
 	if (acpi_enforce_resources == ENFORCE_RESOURCES_NO)
 		return 0;
 	if (!(res->flags & IORESOURCE_IO) && !(res->flags & IORESOURCE_MEM))
