diff -purN linux-5.3.orig/include/linux/mfd/axp20x.h linux-5.3/include/linux/mfd/axp20x.h
--- linux-5.3.orig/include/linux/mfd/axp20x.h	2019-09-15 23:19:32.000000000 +0200
+++ linux-5.3/include/linux/mfd/axp20x.h	2019-09-17 10:09:00.773785973 +0200
@@ -180,6 +180,8 @@ enum axp20x_variants {
 #define AXP20X_BATT_DISCHRG_I_L		0x7d
 #define AXP20X_IPSOUT_V_HIGH_H		0x7e
 #define AXP20X_IPSOUT_V_HIGH_L		0x7f
+#define AXP20X_APS_V_H			0x7e
+#define AXP20X_APS_V_L			0x7f
 
 /* Power supply */
 #define AXP20X_DCDC_MODE		0x80
