diff -urN LPRng-3.8.35.orig/configure LPRng-3.8.35/configure
--- LPRng-3.8.35.orig/configure	2010-11-27 00:49:25.000000000 +0100
+++ LPRng-3.8.35/configure	2012-06-22 23:07:30.000000000 +0200
@@ -14973,7 +14973,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for shell" >&5
 $as_echo_n "checking for shell... " >&6; }
-SHELL=/bin/sh
+SHELL=/bin/bash
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: using $SHELL (FORCED)" >&5
 $as_echo "using $SHELL (FORCED)" >&6; }
 
