diff --git a/src/packages/base/check/base.exp b/src/packages/base/check/base.exp
index 8a70a308..26d73e7d 100644
--- a/src/packages/base/check/base.exp
+++ b/src/packages/base/check/base.exp
@@ -333,7 +333,7 @@ PART_OPT        = 'auto|(RE:PARTITION)|([1-2]?[0-9]{2}|[3-9])[0-9]{2}'
                 : 'invalid PART_OPT, should be auto, a value between
                   300 and 29999 or a partition'
 
-NET_DRV         = '3c509|3c515|3c59x|8139cp|8139too|ac3200|acenic|alx|amd8111e|atl1|atl1c|atl1e|atl2|atp|b44|b53_mdio|be2net|bna|bnx2|bnx2x|cassini|cosa|cs89x0|cxgb|cxgb3|cxgb4|cxgb4vf|de2104x|de4x5|de620|defxx|dl2k|dmfe|dscc4|dummy|e100|e1000|e1000e|ec_bhf|enic|epic100|et131x|farsync|fealnx|fm10k|forcedeth|hamachi|hostess_sv11|hp100|i40e|i40evf|igb|igbvf|ipg|ixgb|ixgbe|ixgbevf|jme|ksz884x|lance|lmc|mlx4_core|mlx4_en|mlx5_core|myri10ge|natsemi|ne|ne2k-pci|ni65|niu|ns83820|pc300too|pch_gbe|pci200syn|pcnet32|qla3xxx|qlcnic|qlge|r6040|r8169|s2io|sb1000|sbni|sc92031|sealevel|seeq8005|sfc|sis190|sis900|skfp|skge|sky2|smc9194|smsc9420|smc-ultra|starfire|stmmac|sundance|sungem|sunhme|tehuti|tg3|tlan|tulip|typhoon|uli526x|via-rhine|via-velocity|vmxnet3|vxge|wanxl|wd|winbond-840|xircom_cb|yellowfin'
+NET_DRV         = '3c509|3c515|3c59x|8139cp|8139too|ac3200|acenic|alx|amd8111e|atl1|atl1c|atl1e|atl2|atp|b44|b53_mdio|be2net|bna|bnx2|bnx2x|cassini|cosa|cs89x0|cxgb|cxgb3|cxgb4|cxgb4vf|de2104x|de4x5|de620|defxx|dl2k|dmfe|dscc4|dummy|e100|e1000|e1000e|ec_bhf|enic|epic100|et131x|farsync|fealnx|fm10k|forcedeth|hamachi|hostess_sv11|hp100|i40e|i40evf|igb|igc|igbvf|ipg|ixgb|ixgbe|ixgbevf|jme|ksz884x|lance|lmc|mlx4_core|mlx4_en|mlx5_core|myri10ge|natsemi|ne|ne2k-pci|ni65|niu|ns83820|pc300too|pch_gbe|pci200syn|pcnet32|qla3xxx|qlcnic|qlge|r6040|r8169|s2io|sb1000|sbni|sc92031|sealevel|seeq8005|sfc|sis190|sis900|skfp|skge|sky2|smc9194|smsc9420|smc-ultra|starfire|stmmac|sundance|sungem|sunhme|tehuti|tg3|tlan|tulip|typhoon|uli526x|via-rhine|via-velocity|vmxnet3|vxge|wanxl|wd|winbond-840|xircom_cb|yellowfin'
                 : 'invalid ethernet driver or not in this KERNEL, please choose one of the drivers in config/base_nic.list'
 +NET_DRV(KERNEL_VERSION=~'^(3\.19|4\.([0-9]|[1-9][0-9]))\..+$') = 'stmmac-pci' : ''
 
diff --git a/src/packages/base/opt/base.txt b/src/packages/base/opt/base.txt
index 41d2ea1c..28a01669 100644
--- a/src/packages/base/opt/base.txt
+++ b/src/packages/base/opt/base.txt
@@ -340,6 +340,7 @@ net_drv_%   i40e        i40e.ko
 net_drv_%   i40evf      i40evf.ko
 net_drv_%   igb         igb.ko
 net_drv_%   igbvf       igbvf.ko
+net_drv_%   igc         igc.ko
 net_drv_%   ipg         ipg.ko
 net_drv_%   ipheth      ipheth.ko
 net_drv_%   ixgb        ixgb.ko
