diff -purN stunnel-5.44.orig/src/network.c stunnel-5.44/src/network.c
--- stunnel-5.44.orig/src/network.c	2017-01-19 09:51:32.000000000 +0100
+++ stunnel-5.44/src/network.c	2018-02-05 20:15:47.303162944 +0100
@@ -508,7 +508,8 @@ int set_socket_options(SOCKET s, int typ
         }
         if(setsockopt(s, ptr->opt_level, ptr->opt_name,
                 (void *)ptr->opt_val[type], opt_size)) {
-            if(get_last_socket_error()==S_EOPNOTSUPP) {
+            int error=get_last_socket_error();
+            if(error==S_EOPNOTSUPP || error==S_ENOPROTOOPT) {
                 /* most likely stdin/stdout or AF_UNIX socket */
                 s_log(LOG_DEBUG,
                     "Option %s not supported on %s socket",
