diff -urN pcmciautils-018.orig/Makefile pcmciautils-018/Makefile
--- pcmciautils-018.orig/Makefile	2013-12-03 20:48:10.000000000 +0100
+++ pcmciautils-018/Makefile	2013-12-03 20:48:37.000000000 +0100
@@ -164,9 +164,9 @@
 	LIB_OBJS = 
 	LIB_PLAIN_OBJS =
 	LIB_PCI_OBJS = -lpci
-	LDFLAGS += -static
+	override LDFLAGS += -static
 else
-	LDFLAGS += -Wl,-warn-common
+	override LDFLAGS += -Wl,-warn-common
 endif
 
 ifeq ($(strip $(V)),false)
