diff -urN pthsem-2.0.8.orig/acinclude.m4 pthsem-2.0.8/acinclude.m4
--- pthsem-2.0.8.orig/acinclude.m4	2009-10-27 08:49:25.000000000 +0100
+++ pthsem-2.0.8/acinclude.m4	2014-11-17 00:40:30.275333705 +0100
@@ -342,8 +342,11 @@
 
 AC_DEFUN(AC_CHECK_DEBUGGING,[dnl
 AC_ARG_ENABLE(debug,dnl
-[  --enable-debug          build for debugging (default=no)],
-[dnl
+[  --enable-debug          build for debugging (default=no)],[dnl
+],[dnl
+enable_debug=no
+])dnl
+if test ".$enable_debug" = .yes; then
 if test ".$ac_cv_prog_gcc" = ".yes"; then
     case $PLATFORM in
         *-*-freebsd*|*-*-solaris* ) CFLAGS="$CFLAGS -pedantic" ;;
@@ -357,9 +360,9 @@
 fi
 msg="enabled"
 AC_DEFINE(PTH_DEBUG, 1, [define to enable Pth debugging])
-],[
+else
 msg="disabled"
-])dnl
+fi
 AC_MSG_CHECKING(for compilation debug mode)
 AC_MSG_RESULT([$msg])
 if test ".$msg" = .enabled; then
