diff -purN serdisplib-2.01.orig/src/Makefile.in serdisplib-2.01/src/Makefile.in
--- serdisplib-2.01.orig/src/Makefile.in	2017-02-02 13:37:48.000000000 +0100
+++ serdisplib-2.01/src/Makefile.in	2018-02-05 17:38:40.607224741 +0100
@@ -188,7 +188,7 @@ $(LIB_DIR)/$(LIB_SHARED): $(LIB_OBJECTS)
 	cd $(LIB_DIR) && $(LN_S) -f $(LIB_SHARED) lib$(LIB_NAME).so.$(VERSION_MAJOR)
 	cd $(LIB_DIR) && $(LN_S) -f lib$(LIB_NAME).so.$(VERSION_MAJOR) lib$(LIB_NAME).so
 
-testserdisp: $(LIB_DIR)/$(LIB_STATIC) $(OBJECTS)
+testserdisp: $(LIB_DIR)/$(LIB_STATIC) $(LIB_DIR)/$(LIB_SHARED) $(OBJECTS)
 	$(CC) -o testserdisp $(OBJECTS) $(LIB_SERDISP) $(EXTRA_LIBS_STATIC)
 
 clean:
