diff -urN serdisplib-1.97.9.orig/configure.in serdisplib-1.97.9/configure.in
--- serdisplib-1.97.9.orig/configure.in	2010-02-21 20:49:29.000000000 +0100
+++ serdisplib-1.97.9/configure.in	2015-05-28 23:06:26.637862190 +0200
@@ -148,6 +148,7 @@
 has_libSDL="false"   # pre-init
 if test ! -z "${LIBSDL_CONFIG}"; then
   LIBSDL_CFLAGS=`${LIBSDL_CONFIG} --cflags`
+  TEMP_CFLAGS="${CFLAGS}"
   CFLAGS="${CFLAGS} ${LIBSDL_CFLAGS}"
 
   #defines either HAVE_SDL_SDL_H or HAVE_SDL_H
@@ -182,6 +183,10 @@
         fi
       fi
   fi
+
+  if test "$has_libSDL" != "true"; then
+    CFLAGS="${TEMP_CFLAGS}"
+  fi
 fi
 AC_SUBST(has_libSDL)
 AC_SUBST(HAVE_LIBSDL)
