diff -purN serdisplib-2.01.orig/configure.in serdisplib-2.01/configure.in
--- serdisplib-2.01.orig/configure.in	2017-02-05 00:04:53.000000000 +0100
+++ serdisplib-2.01/configure.in	2018-02-05 18:41:20.901893695 +0100
@@ -61,7 +61,7 @@ AC_DEFUN(MY_ARG_WITH_LIST,
 # eg: MY_GET_FALLBACK_SONAME ([usb], [libusb-0])
 AC_DEFUN(MY_GET_FALLBACK_SONAME,
 [
- _SONAME_val=`( echo 'int main(){;}' | $CC -O0 -o /tmp/conftest.$ac_ext.bin -xc - -Wl,--no-as-needed -l$1 ) 2> /dev/null && ( ldd /tmp/conftest.$ac_ext.bin | egrep '$2' | awk '{print [$]1;exit}' ) `
+ _SONAME_val=`( echo 'int main(){;}' | $CC -O0 -o /tmp/conftest.$ac_ext.bin -xc - -Wl,--no-as-needed -l$1 ) 2> /dev/null && ( readelf -d /tmp/conftest.$ac_ext.bin | sed -n -r '/$2/s/^.*\@<:@($2.*)\@:>@.*$/\1/p' ) `
  _DUMMY_RC=`rm -f /tmp/conftest.$ac_ext.bin 2> /dev/null`
  if test -n "${_SONAME_val}"; then
    AC_DEFINE_UNQUOTED(FALLBACK_SONAME_lib$1, "${_SONAME_val}", fallback soname for lib$1)
