diff -purN collectd-5.7.1.orig/src/daemon/plugin.c collectd-5.7.1/src/daemon/plugin.c
--- collectd-5.7.1.orig/src/daemon/plugin.c	2017-01-23 08:53:57.700448666 +0100
+++ collectd-5.7.1/src/daemon/plugin.c	2017-08-30 16:42:05.645966084 +0200
@@ -1033,15 +1033,17 @@ int plugin_load(char const *plugin_name,
       continue;
     }
 
-    if (lstat(filename, &statbuf) == -1) {
+    if (stat(filename, &statbuf) == -1) {
       char errbuf[1024];
       WARNING("plugin_load: stat (\"%s\") failed: %s", filename,
               sstrerror(errno, errbuf, sizeof(errbuf)));
       continue;
+#if 0
     } else if (!S_ISREG(statbuf.st_mode)) {
       /* don't follow symlinks */
       WARNING("plugin_load: %s is not a regular file.", filename);
       continue;
+#endif
     }
 
     status = plugin_load_file(filename, flags);
