diff -purN collectd-5.7.1.orig/src/netlink.c collectd-5.7.1/src/netlink.c
--- collectd-5.7.1.orig/src/netlink.c	2017-01-23 08:53:57.716449156 +0100
+++ collectd-5.7.1/src/netlink.c	2017-08-30 16:43:11.677039488 +0200
@@ -480,7 +480,7 @@ static int qos_filter_cb(const struct nl
     if (strcmp(tc_type, "filter") == 0)
       numberic_id = tm->tcm_parent;
 
-    snprintf(tc_inst, sizeof(tc_inst), "%s-%x:%x", kind, numberic_id >> 16,
+    snprintf(tc_inst, sizeof(tc_inst), "%s-%x_%x", kind, numberic_id >> 16,
              numberic_id & 0x0000FFFF);
   }
