diff -purN collectd-5.7.1.orig/src/onewire.c collectd-5.7.1/src/onewire.c
--- collectd-5.7.1.orig/src/onewire.c	2017-01-23 08:53:57.716449156 +0100
+++ collectd-5.7.1/src/onewire.c	2017-08-30 16:56:41.953647164 +0200
@@ -31,7 +31,7 @@
 #include <sys/types.h>
 
 #define OW_FAMILY_LENGTH 8
-#define OW_FAMILY_MAX_FEATURES 2
+#define OW_FAMILY_MAX_FEATURES 7
 struct ow_family_features_s {
   char family[OW_FAMILY_LENGTH];
   struct {
@@ -72,6 +72,96 @@ static ow_family_features_t ow_family_fe
        /* type = */ "temperature",
        /* type_instance = */ ""}},
      /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "1E.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "21.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "2E.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "30.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "32.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "35.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "3B.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "3D.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "41.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "42.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "51.",
+     {{/* filename = */ "temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""}},
+     /* features_num = */ 1},
+    {/* ??? */
+     /* family = */ "7E.",
+     {{/* filename = */ "EDS0068/temperature",
+       /* type = */ "temperature",
+       /* type_instance = */ ""},
+      {/* filename = */ "EDS0068/humidity",
+       /* type = */ "humidity",
+       /* type_instance = */ ""},
+      {/* filename = */ "EDS0068/humidex",
+       /* type = */ "humidex",
+       /* type_instance = */ ""},
+      {/* filename = */ "EDS0068/dew_point",
+       /* type = */ "dew_point",
+       /* type_instance = */ ""},
+      {/* filename = */ "EDS0068/heat_index",
+       /* type = */ "heat_index",
+       /* type_instance = */ ""},
+      {/* filename = */ "EDS0068/light",
+       /* type = */ "illuminance",
+       /* type_instance = */ ""},
+      {/* filename = */ "EDS0068/pressure",
+       /* type = */ "pressure",
+       /* type_instance = */ ""}},
+     /* features_num = */ 7},
     {/* DS2436 Volts/Temp */
      /* family = */ "1B.",
      {{/* filename = */ "temperature",
@@ -82,8 +172,20 @@ static ow_family_features_t ow_family_fe
      /* family = */ "26.",
      {{/* filename = */ "temperature",
        /* type = */ "temperature",
+       /* type_instance = */ ""},
+      {/* filename = */ "humidity",
+       /* type = */ "humidity",
+       /* type_instance = */ ""},
+      {/* filename = */ "VAD",
+       /* type = */ "voltage",
+       /* type_instance = */ ""},
+      {/* filename = */ "S3-R1-A/illuminance",
+       /* type = */ "illuminance",
+       /* type_instance = */ ""},
+      {/* filename = */ "B1-R1-A/pressure",
+       /* type = */ "pressure",
        /* type_instance = */ ""}},
-     /* features_num = */ 1}};
+     /* features_num = */ 5}};
 static int ow_family_features_num = STATIC_ARRAY_SIZE(ow_family_features);
 
 static char *device_g = NULL;
