diff -purN collectd-5.8.0.orig/Makefile.am collectd-5.8.0/src/Makefile.am
--- collectd-5.8.0.orig/Makefile.am	2017-01-23 08:53:57.688448300 +0100
+++ collectd-5.8.0/Makefile.am	2017-08-30 16:58:28.082154800 +0200
@@ -871,7 +871,7 @@ endif
 if BUILD_PLUGIN_ONEWIRE
 pkglib_LTLIBRARIES += onewire.la
 onewire_la_SOURCES = src/onewire.c
-onewire_la_CPPFLAGS = $(AM_CPPFLAGS) $(BUILD_WITH_LIBOWCAPI_CPPFLAGS)
+onewire_la_CPPFLAGS = $(AM_CPPFLAGS) $(BUILD_WITH_LIBOWCAPI_CPPFLAGS) -D_GNU_SOURCE
 onewire_la_LDFLAGS = $(PLUGIN_LDFLAGS) $(BUILD_WITH_LIBOWCAPI_LDFLAGS)
 onewire_la_LIBADD = libignorelist.la $(BUILD_WITH_LIBOWCAPI_LIBS)
 endif
