diff -urN mc-4.8.7.orig/src/textconf.c mc-4.8.7/src/textconf.c
--- mc-4.8.7.orig/src/textconf.c	2012-12-10 22:04:32.000000000 +0100
+++ mc-4.8.7/src/textconf.c	2012-12-31 23:51:33.697241184 +0100
@@ -229,7 +229,7 @@
 void
 show_configure_options (void)
 {
-    (void) printf ("%s\n", MC_CONFIGURE_ARGS);
+    (void) printf ("\n");
 }
 
 /* --------------------------------------------------------------------------------------------- */
