diff -urN yadifa-2.0.0-4192.org/bin/yadifa/yadifa-config.c yadifa-2.0.0-4192.fli4l/bin/yadifa/yadifa-config.c
--- yadifa-2.0.0-4192.org/bin/yadifa/yadifa-config.c	2014-09-05 15:02:05.000000000 +0200
+++ yadifa-2.0.0-4192.fli4l/bin/yadifa/yadifa-config.c	2014-10-23 12:15:02.000000000 +0200
@@ -311,10 +311,10 @@
             osformatln(termout, "%s %s (%s)", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE);
             break;
         case 2:
-            osformatln(termout, "%s %s (released %s, compiled %s)", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE, COMPILEDATE);
+            osformatln(termout, "%s %s (released %s, compiled %s)", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE, RELEASEDATE);
             break;
         case 3:
-            osformatln(termout, "%s %s (released %s, compiled %s)", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE, COMPILEDATE);
+            osformatln(termout, "%s %s (released %s, compiled %s)", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE, RELEASEDATE);
             yadifa_print_authors();
             break;
         default:
diff -urN yadifa-2.0.0-4192.org/sbin/yadifad/confs.c yadifa-2.0.0-4192.fli4l/sbin/yadifad/confs.c
--- yadifa-2.0.0-4192.org/sbin/yadifad/confs.c	2014-09-05 15:02:02.000000000 +0200
+++ yadifa-2.0.0-4192.fli4l/sbin/yadifad/confs.c	2014-10-23 12:14:32.000000000 +0200
@@ -227,10 +227,10 @@
 	    osformatln(termout, "%s %s (%s)\n", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE);
 	    break;
 	case 2:
-	    osformatln(termout, "%s %s (released %s, compiled %s)\n\nbuild settings: %s\n", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE, COMPILEDATE, BUILD_OPTIONS);
+	    osformatln(termout, "%s %s (released %s, compiled %s)\n\nbuild settings: %s\n", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE, RELEASEDATE, BUILD_OPTIONS);
 	    break;
         case 3:
-	    osformatln(termout, "%s %s (released %s, compiled %s)\n", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE, COMPILEDATE);
+	    osformatln(termout, "%s %s (released %s, compiled %s)\n", PROGRAM_NAME, PROGRAM_VERSION, RELEASEDATE, RELEASEDATE);
             yadifad_print_authors();
             break;
 	default:
