diff -urN yadifa-2.0.5-4735.orig/m4/eurid.m4 yadifa-2.0.5-4735/m4/eurid.m4
--- yadifa-2.0.5-4735.orig/m4/eurid.m4	2015-03-03 11:42:47.324288352 +0100
+++ yadifa-2.0.5-4735/m4/eurid.m4	2015-05-04 19:10:45.168267713 +0200
@@ -267,6 +267,7 @@
 CPU_UNKNOWN=1
 
 CFLAGS3264=
+if test yes != "$cross_compiling"; then
 case "$(uname -i 2>/dev/null)" in
 	SUNW,SPARC-Enterprise-T1000)
 		AC_DEFINE_UNQUOTED([HAS_CPU_NIAGARA], [1], [T1000 has a Niagara cpu])
@@ -292,6 +293,7 @@
 	*)
 		;;
 esac
+fi
 
 case "${CPU_UNKNOWN}" in
 	1)
