diff -purN olsrd-0.9.6.1.orig/make/hash_source.sh olsrd-0.9.6.1/make/hash_source.sh
--- olsrd-0.9.6.1.orig/make/hash_source.sh	2017-02-10 11:20:11.000000000 +0100
+++ olsrd-0.9.6.1/make/hash_source.sh	2018-01-29 10:36:36.023114591 +0100
@@ -75,7 +75,7 @@ fi
 gitDescriptor="$(git describe --dirty --always 2> /dev/null)"
 
 sourceHash="$(cat $(find . -name *.[ch] | grep -v -E '[/\\]?builddata.c$') | "$md5Command" | awk '{ print $1; }')"
-hostName="$(hostname)"
+hostName="(unknown)"
 buildDate="$(date +"%Y-%m-%d %H:%M:%S")"
 
 tmpBuildDataTxt="$(mktemp -t olsrd.hash_source.XXXXXXXXXX)"
