diff -urN crosstool-ng-1.19.0.orig/steps.mk crosstool-ng-1.19.0/steps.mk
--- crosstool-ng-1.19.0.orig/steps.mk	2013-10-02 20:06:23.000000000 +0200
+++ crosstool-ng-1.19.0/steps.mk	2014-01-02 09:16:30.708051483 +0100
@@ -23,9 +23,9 @@
             binutils_for_build       \
             companion_libs_for_host  \
             binutils_for_host        \
+            libc_start_files         \
             cc_core_pass_1           \
             kernel_headers           \
-            libc_start_files         \
             cc_core_pass_2           \
             libc                     \
             cc_for_build             \
