diff -urN lptkeys.orig/main.c lptkeys/main.c
--- lptkeys.orig/main.c	2001-07-04 20:28:15.000000000 +0200
+++ lptkeys/main.c	2012-07-25 15:46:46.000000000 +0200
@@ -84,18 +84,18 @@
 
 void usage ()
 {
-  printf ("\nUsage: lptkeys [-5] [-c] [-f] [-h] [-v] [-n] [-m 0x..] [-p 0x378|0x278|0x3bc]\n
-                -5 set +5V on pin 17 (SELECT PRINTER)
-                   without this switch: don't touch pin 17
-                -c shows an example circuit
-                -f report current status - do not wait for key pressed
-                -h shows help and exits
-                -m mask the lines according to 0x.. 
-                   the bitmask accords to the default output
-                   format
-                -v shows version and exits
-                -p use given IO port
-                -n output key as number rather then binary\n\n");
+  printf ("\nUsage: lptkeys [-5] [-c] [-f] [-h] [-v] [-n] [-m 0x..] [-p 0x378|0x278|0x3bc]\n"
+"                -5 set +5V on pin 17 (SELECT PRINTER)\n"
+"                   without this switch: don't touch pin 17\n"
+"                -c shows an example circuit\n"
+"                -f report current status - do not wait for key pressed\n"
+"                -h shows help and exits\n"
+"                -m mask the lines according to 0x.. \n"
+"                   the bitmask accords to the default output\n"
+"                   format\n"
+"                -v shows version and exits\n"
+"                -p use given IO port\n"
+"                -n output key as number rather then binary\n\n");
 }
 
 void version ()
@@ -105,39 +105,39 @@
 
 void circuit ()
 {
-  printf ("\nDB25  Pinname                5 x 4,7k
-                             _____
-15    ERROR    ---------o----|___|--------o +5V
-                        /
-                        o
-               S1       |
-                        -
-                             _____
-13    SLCT     ---------o----|___|--------o +5V
-                        /
-                        o
-               S2       |
-                        -
-                             _____
-12    PE       ---------o----|___|--------o +5V
-                        /
-                        o
-               S3       |
-                        -
-                             _____
-10    ACK      ---------o----|___|--------o +5V
-                        /
-                        o
-               S4       |
-                        -
-                             _____
-11    BUSY     ---------o----|___|--------o +5V
-                        /
-                        o
-               S5       |                 ^
-                        -                 |
-                                          |
-17    SLCT PRI ---------------------------+ +5V \n\n");
+  printf ("\nDB25  Pinname                5 x 4,7k\n"
+"                             _____\n"
+"15    ERROR    ---------o----|___|--------o +5V\n"
+"                        /\n"
+"                        o\n"
+"               S1       |\n"
+"                        -\n"
+"                             _____\n"
+"13    SLCT     ---------o----|___|--------o +5V\n"
+"                        /\n"
+"                        o\n"
+"               S2       |\n"
+"                        -\n"
+"                             _____\n"
+"12    PE       ---------o----|___|--------o +5V\n"
+"                        /\n"
+"                        o\n"
+"               S3       |\n"
+"                        -\n"
+"                             _____\n"
+"10    ACK      ---------o----|___|--------o +5V\n"
+"                        /\n"
+"                        o\n"
+"               S4       |\n"
+"                        -\n"
+"                             _____\n"
+"11    BUSY     ---------o----|___|--------o +5V\n"
+"                        /\n"
+"                        o\n"
+"               S5       |                 ^\n"
+"                        -                 |\n"
+"                                          |\n"
+"17    SLCT PRI ---------------------------+ +5V \n\n");
 
   printf ("NO WARRANTY FOR DAMAGES OF YOUR HARDWARE!!!\n\n");
 }
@@ -191,9 +191,9 @@
         exit (-1);
       }
       if (Port!=0x278 && Port!=0x378 && Port!=0x3bc) {
-        fprintf (stderr, "lptkeys: Port may only be 0x378 (lpt1)
-                          0x278 (lpt2)
-                          0x3bc (lpt3)!\n");
+        fprintf (stderr, "lptkeys: Port may only be 0x378 (lpt1)\n"
+                         "0x278 (lpt2)\n"
+                         "0x3bc (lpt3)!\n");
         exit (-1);
       }
       break;
