diff -urN pciutils-3.2.0.orig/Makefile pciutils-3.2.0/Makefile
--- pciutils-3.2.0.orig/Makefile	2013-04-19 14:58:49.000000000 +0200
+++ pciutils-3.2.0/Makefile	2013-11-03 20:27:45.370809221 +0100
@@ -130,7 +130,7 @@
 
 install-pcilib: lib/$(PCILIB)
 	$(DIRINSTALL) -m 755 $(DESTDIR)$(LIBDIR)
-	$(INSTALL) -c -m 644 lib/$(PCILIB) $(DESTDIR)$(LIBDIR)
+	$(INSTALL) -c -m 755 lib/$(PCILIB) $(DESTDIR)$(LIBDIR)
 
 install-lib: $(PCIINC_INS) lib/$(PCILIBPC) install-pcilib
 	$(DIRINSTALL) -m 755 $(DESTDIR)$(INCDIR)/pci $(DESTDIR)$(PKGCFDIR)
