diff -urN socat-2.0.0-b8.orig/socat.c socat-2.0.0-b8/socat.c
--- socat-2.0.0-b8.orig/socat.c	2015-04-06 11:52:38.000000000 +0200
+++ socat-2.0.0-b8/socat.c	2015-05-14 17:23:15.757519782 +0200
@@ -50,7 +50,6 @@
 static const char socatversion[] =
 #include "./VERSION"
       ;
-static const char timestamp[] = BUILD_DATE;
 
 const char copyright_socat[] = "socat by Gerhard Rieger - see www.dest-unreach.org";
 #if WITH_OPENSSL
@@ -274,7 +273,7 @@
    Info(copyright_openssl);
    Info(copyright_ssleay);
 #endif
-   Debug2("socat version %s on %s", socatversion, timestamp);
+   Debug1("socat version %s", socatversion);
    xiosetenv("VERSION", socatversion, 1, NULL);	/* SOCAT_VERSION */
    uname(&ubuf);	/* ! here we circumvent internal tracing (Uname) */
    Debug4("running on %s version %s, release %s, machine %s\n",
@@ -389,7 +388,7 @@
    struct utsname ubuf;
 
    fputs(copyright_socat, fd); fputc('\n', fd);
-   fprintf(fd, "socat version %s on %s\n", socatversion, timestamp);
+   fprintf(fd, "socat version %s\n", socatversion);
    Uname(&ubuf);
    fprintf(fd, "   running on %s version %s, release %s, machine %s\n",
 	   ubuf.sysname, ubuf.version, ubuf.release, ubuf.machine);
