diff -urN uClibc++-0.2.4.orig/src/Makefile uClibc++-0.2.4/src/Makefile
--- uClibc++-0.2.4.orig/src/Makefile	2012-05-25 23:15:31.000000000 +0200
+++ uClibc++-0.2.4/src/Makefile	2013-11-30 11:56:21.456916457 +0100
@@ -3,7 +3,7 @@
 
 WR_CXX := WRAPPER_INCLUDEDIR=-I$(TOPDIR)include $(TOPDIR)bin/g++-uc
 
-LDFLAGS += -nodefaultlibs -shared -Wl,-soname,$(SHARED_MAJORNAME) $(BUILD_EXTRA_LIBRARIES)
+override LDFLAGS += -nodefaultlibs -shared -Wl,-soname,$(SHARED_MAJORNAME) $(BUILD_EXTRA_LIBRARIES)
 ifneq ($(DODEBUG),y)
 LDFLAGS += -Wl,-s
 endif
