diff -urN lib.orig/regcomp.c lib/regcomp.c
--- lib.orig/regcomp.c	2012-09-08 08:34:56.000000000 +0200
+++ lib/regcomp.c	2012-09-08 21:47:08.815523659 +0200
@@ -898,12 +898,14 @@
   dfa->map_notascii = (_NL_CURRENT_WORD (LC_CTYPE, _NL_CTYPE_MAP_TO_NONASCII)
 		       != 0);
 #else
+#ifndef _WIN32
   codeset_name = nl_langinfo (CODESET);
   if ((codeset_name[0] == 'U' || codeset_name[0] == 'u')
       && (codeset_name[1] == 'T' || codeset_name[1] == 't')
       && (codeset_name[2] == 'F' || codeset_name[2] == 'f')
       && strcmp (codeset_name + 3 + (codeset_name[3] == '-'), "8") == 0)
     dfa->is_utf8 = 1;
+#endif
 
   /* We check exhaustively in the loop below if this charset is a
      superset of ASCII.  */
diff -urN lib.orig/regex_internal.h lib/regex_internal.h
--- lib.orig/regex_internal.h	2012-09-08 21:47:34.846031703 +0200
+++ lib/regex_internal.h	2012-09-08 21:47:08.816523678 +0200
@@ -25,7 +25,9 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifndef _WIN32
 #include <langinfo.h>
+#endif
 #include <locale.h>
 #include <wchar.h>
 #include <wctype.h>
@@ -446,10 +448,9 @@
 #define re_string_skip_bytes(pstr,idx) ((pstr)->cur_idx += (idx))
 #define re_string_set_index(pstr,idx) ((pstr)->cur_idx = (idx))
 
-#include <alloca.h>
-
 #ifndef _LIBC
 # if HAVE_ALLOCA
+#  include <alloca.h>
 /* The OS usually guarantees only one guard page at the bottom of the stack,
    and a page size can be as small as 4096 bytes.  So we cannot safely
    allocate anything larger than 4096 bytes.  Also care for the possibility
