diff -urN lib.orig/regex.c lib/regex.c
--- lib.orig/regex.c	2012-09-08 08:34:56.000000000 +0200
+++ lib/regex.c	2012-09-09 16:18:09.311331176 +0200
@@ -17,7 +17,7 @@
    with this program; if not, see <http://www.gnu.org/licenses/>.  */
 
 #ifndef _LIBC
-# include <config.h>
+# include <regex-config.h>
 
 # if (__GNUC__ == 4 && 6 <= __GNUC_MINOR__) || 4 < __GNUC__
 #  pragma GCC diagnostic ignored "-Wsuggest-attribute=pure"
