diff -purN linux-5.3.orig/drivers/net/ethernet/sun/sunhme.c linux-5.3/drivers/net/ethernet/sun/sunhme.c
--- linux-5.3.orig/drivers/net/ethernet/sun/sunhme.c	2019-09-15 23:19:32.000000000 +0200
+++ linux-5.3/drivers/net/ethernet/sun/sunhme.c	2019-09-17 10:07:32.809994484 +0200
@@ -704,8 +704,6 @@ static void happy_meal_timer(struct time
 			/* Enter force mode. */
 	do_force_mode:
 			hp->sw_bmcr = happy_meal_tcvr_read(hp, tregs, MII_BMCR);
-			printk(KERN_NOTICE "%s: Auto-Negotiation unsuccessful, trying force link mode\n",
-			       hp->dev->name);
 			hp->sw_bmcr = BMCR_SPEED100;
 			happy_meal_tcvr_write(hp, tregs, MII_BMCR, hp->sw_bmcr);
 
@@ -819,10 +817,6 @@ static void happy_meal_timer(struct time
 					 * chip and try all over again.
 					 */
 
-					/* Let the user know... */
-					printk(KERN_NOTICE "%s: Link down, cable problem?\n",
-					       hp->dev->name);
-
 					ret = happy_meal_init(hp);
 					if (ret) {
 						/* ho hum... */
