diff -urN inetutils-1.9.1/Makefile.in inetutils-1.9.1.patched/Makefile.in
--- inetutils-1.9.1/Makefile.in	2012-01-06 15:11:46.000000000 +0100
+++ inetutils-1.9.1.patched/Makefile.in	2012-03-30 13:43:20.000000000 +0200
@@ -1202,7 +1202,7 @@
 SUBDIRS = lib \
 	libinetutils libtelnet libicmp libls \
 	src telnet telnetd ftp ftpd talk talkd whois ping ifconfig \
-	doc man \
+	doc \
 	tests
 
 DISTCLEANFILES = pathdefs.make paths.defs $(PACKAGE)-$(VERSION).tar.gz
