diff -purN openvpn-2.4.0.orig/src/openvpn/options.c openvpn-2.4.0/src/openvpn/options.c
--- openvpn-2.4.0.orig/src/openvpn/options.c	2016-12-26 12:51:00.000000000 +0100
+++ openvpn-2.4.0/src/openvpn/options.c	2017-08-30 21:32:58.913434850 +0200
@@ -107,7 +107,6 @@ const char title_string[] =
 #ifdef HAVE_AEAD_CIPHER_MODES
     " [AEAD]"
 #endif
-    " built on " __DATE__
 ;
 
 #ifndef ENABLE_SMALL
