diff -urN libpcap-1.7.2.orig/pcap-config.in libpcap-1.7.2/pcap-config.in
--- libpcap-1.7.2.orig/pcap-config.in	2014-11-13 09:58:52.000000000 +0100
+++ libpcap-1.7.2/pcap-config.in	2015-03-29 17:09:59.262110112 +0200
@@ -36,16 +36,6 @@
 	esac
 	shift
 done
-if [ "$V_RPATH_OPT" != "" ]
-then
-	#
-	# If libdir isn't /usr/lib, add it to the run-time linker path.
-	#
-	if [ "$libdir" != "/usr/lib" ]
-	then
-		RPATH=$V_RPATH_OPT$libdir
-	fi
-fi
 if [ "$static" = 1 ]
 then
 	#
