diff -urN busybox-1.19.4/Makefile.flags busybox-1.19.4.patched/Makefile.flags
--- busybox-1.19.4/Makefile.flags	2012-02-04 20:34:24.000000000 +0100
+++ busybox-1.19.4.patched/Makefile.flags	2012-03-14 18:02:35.000000000 +0100
@@ -15,7 +15,7 @@
 	-include include/autoconf.h \
 	-D_GNU_SOURCE -DNDEBUG \
 	$(if $(CONFIG_LFS),-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64) \
-	-D"BB_VER=KBUILD_STR($(BB_VER))" -DBB_BT=AUTOCONF_TIMESTAMP
+	-D"BB_VER=KBUILD_STR($(BB_VER))" -D"BB_BT=KBUILD_STR(fli4l)"
 
 CFLAGS += $(call cc-option,-Wall,)
 CFLAGS += $(call cc-option,-Wshadow,)
