diff -purN busybox-1.25.0.orig/sysklogd/syslogd.c busybox-1.25.0/sysklogd/syslogd.c
--- busybox-1.25.0.orig/sysklogd/syslogd.c	2016-05-26 19:42:45.000000000 +0200
+++ busybox-1.25.0/sysklogd/syslogd.c	2016-08-17 22:36:40.379239453 +0200
@@ -444,7 +444,7 @@ static void parse_syslogdcfg(const char
 
 			/* parse priority */
 			if (*t == '*')
-				primap = 0xff; /* all 8 log levels enabled */
+				primap = negated_prio ? 0 : 0xff; /* all 8 log levels enabled */
 			else {
 				uint8_t priority;
 				code = find_by_name(t, prioritynames);
