diff -purN busybox-1.26.2.orig/coreutils/readlink.c busybox-1.26.2/coreutils/readlink.c
--- busybox-1.26.2.orig/coreutils/readlink.c	2016-12-10 18:46:36.000000000 +0100
+++ busybox-1.26.2/coreutils/readlink.c	2017-08-30 14:19:32.319417000 +0200
@@ -20,7 +20,7 @@
 //config:	help
 //config:	  Enable the readlink option (-f).
 
-//applet:IF_READLINK(APPLET(readlink, BB_DIR_USR_BIN, BB_SUID_DROP))
+//applet:IF_READLINK(APPLET_NOFORK(readlink, readlink, BB_DIR_USR_BIN, BB_SUID_DROP, readlink))
 
 //kbuild:lib-$(CONFIG_READLINK) += readlink.o
 
