diff -urN lcd4linux-1191.orig/drv_dpf.c lcd4linux-1191/drv_dpf.c
--- lcd4linux-1191.orig/drv_dpf.c	2012-01-16 03:50:03.000000000 +0100
+++ lcd4linux-1191/drv_dpf.c	2012-10-31 16:49:48.247383596 +0100
@@ -73,15 +73,13 @@
     int error;
     char *dev;
 
-    // Currently, the Port specification is unused.
-
     dev = cfg_get(section, "Port", NULL);
     if (dev == NULL || *dev == '\0') {
 	error("dpf: no '%s.Port' entry from %s", section, cfg_source());
 	return -1;
     }
 
-    error = dpf_open(NULL, &g_h);
+    error = dpf_open(dev, &g_h);
     if (error < 0) {
 	error("dpf: cannot open dpf device %s", dev);
 	return -1;
@@ -107,7 +105,6 @@
 {
     int r, c;
     short rect[4];
-    unsigned long i;
     RGBA p;
     unsigned char *pix;
 
