diff -urN lcd4linux-1191.orig/plugins.m4 lcd4linux-1191/plugins.m4
--- lcd4linux-1191.orig/plugins.m4	2011-12-26 07:28:05.000000000 +0100
+++ lcd4linux-1191/plugins.m4	2013-02-11 09:26:34.063909105 +0100
@@ -400,7 +400,7 @@
 
 # IConv
 if test "$PLUGIN_ICONV" = "yes"; then
-   AM_ICONV
+   m4_pattern_allow([AM_ICONV])
    if test "$am_cv_func_iconv" = "yes"; then 
       PLUGINS="$PLUGINS plugin_iconv.o"
       PLUGINLIBS="$PLUGINLIBS $LIBICONV"
