diff -urN lcd4linux-1191.orig/drivers.m4 lcd4linux-1191/drivers.m4
--- lcd4linux-1191.orig/drivers.m4	2012-04-09 06:46:59.000000000 +0200
+++ lcd4linux-1191/drivers.m4	2012-07-27 22:35:35.297610217 +0200
@@ -381,7 +381,7 @@
    if test "$has_libdpf" = "true"; then
       GRAPHIC="yes"
       DRIVERS="$DRIVERS drv_dpf.o"
-      DRVLIBS="$DRVLIBS -Llibdpf -ldpf -lusb"
+      DRVLIBS="$DRVLIBS -ldpf -lusb"
       AC_DEFINE(WITH_DPF,1,[DPF driver])
    else
       AC_MSG_WARN(libdpf.h not found: DPF driver disabled)
@@ -537,7 +537,7 @@
    if test "$has_luise" = "true"; then
       GRAPHIC="yes"
       DRIVERS="$DRIVERS drv_LUIse.o"
-      DRVLIBS="$DRVLIBS -L/usr/local/lib -lluise"
+      DRVLIBS="$DRVLIBS -lluise"
       AC_DEFINE(WITH_LUISE,1,[LUIse driver])
    else
       AC_MSG_WARN(luise.h not found: LUIse driver disabled)
@@ -741,7 +741,7 @@
    if test "$has_serdisplib" = "true"; then
       GRAPHIC="yes"
       DRIVERS="$DRIVERS drv_serdisplib.o"
-      DRVLIBS="$DRVLIBS -L/usr/local/lib -lserdisp"
+      DRVLIBS="$DRVLIBS -lserdisp"
       AC_DEFINE(WITH_SERDISPLIB,1,[serdisplib driver])
       if test "$has_usb" = "true"; then
          LIBUSB="yes"
@@ -774,7 +774,7 @@
    if test "$has_st2205" = "true"; then
       GRAPHIC="yes"
       DRIVERS="$DRIVERS drv_st2205.o"
-      DRVLIBS="$DRVLIBS -L/usr/local/lib -lst2205"
+      DRVLIBS="$DRVLIBS -lst2205"
       AC_DEFINE(WITH_ST2205,1,[st2205 driver])
    else
       AC_MSG_WARN(st2205.h not found: st2205 driver disabled)
@@ -849,7 +849,7 @@
       GRAPHIC="yes"
       KEYPAD="yes"      
       DRIVERS="$DRIVERS drv_vnc.o"
-      DRVLIBS="$DRVLIBS -L/usr/local/lib -lvncserver -lz"
+      DRVLIBS="$DRVLIBS -lvncserver -lz"
       AC_DEFINE(WITH_VNC,1,[vnc driver])
    else
       AC_MSG_WARN(libvncserver not found: vnc driver disabled)
