diff -urN lcd4linux-1191.orig/Makefile.am lcd4linux-1191/Makefile.am
--- lcd4linux-1191.orig/Makefile.am	2012-02-22 04:11:31.000000000 +0100
+++ lcd4linux-1191/Makefile.am	2012-08-01 00:59:21.833604417 +0200
@@ -13,7 +13,7 @@
 # Fixme: -W should be renamed to -Wextra someday...
 AM_CFLAGS = -D_GNU_SOURCE -Wall -Wextra -fno-strict-aliasing
 
-LIBTOOL=libtool
+LIBTOOL=@LIBTOOL@
 ACLOCAL_AMFLAGS=-I m4
 # use this for lots of warnings
 #AM_CFLAGS = -D_GNU_SOURCE -std=c99 -m64 -Wall -W -pedantic -Wno-variadic-macros -fno-strict-aliasing
