diff -urN lcd4linux-1191.orig/plugin_netinfo.c lcd4linux-1191/plugin_netinfo.c
--- lcd4linux-1191.orig/plugin_netinfo.c	2010-11-20 21:22:05.000000000 +0100
+++ lcd4linux-1191/plugin_netinfo.c	2015-05-04 20:57:00.524541885 +0200
@@ -263,8 +263,12 @@
 
     sin = get_netmask(arg1);
     if (NULL != sin) {
-	logval = (long double) (get_netmask(arg1)->sin_addr.s_addr);
-	netlen = (int) rint(log2l(logval) / log2l(2.0));
+	in_addr_t mask = 0x80000000;
+	in_addr_t addr = sin->sin_addr.s_addr;
+	while ((addr & mask) != 0) {
+	    ++netlen;
+	    mask >>= 1;
+	}
 	qprintf(value, sizeof(value), "/%d", netlen);
     } else {
 	qprintf(value, sizeof(value), "/?");
