diff -urN lbase64-20120820-1.orig/base64/Makefile lbase64-20120820-1/base64/Makefile
--- lbase64-20120820-1.orig/base64/Makefile	2012-08-09 05:01:56.000000000 +0200
+++ lbase64-20120820-1/base64/Makefile	2013-11-30 09:20:30.316828768 +0100
@@ -17,7 +17,7 @@
 CFLAGS= $(INCS) $(WARN) -O2 $G
 WARN= -ansi -pedantic -Wall -Wextra
 INCS= -I$(LUAINC)
-MAKESO= $(CC) -shared
+MAKESO= $(CC) $(LDFLAGS) -shared
 #MAKESO= $(CC) -bundle -undefined dynamic_lookup
 
 MYNAME= base64
