diff -purN dnsmasq-2.86.orig/src/dnsmasq.h dnsmasq-2.86/src/dnsmasq.h
--- dnsmasq-2.86.orig/src/dnsmasq.h	2021-09-08 22:21:22.000000000 +0200
+++ dnsmasq-2.86/src/dnsmasq.h	2022-07-17 08:39:23.958471099 +0200
@@ -1580,7 +1580,7 @@ void ubus_event_bcast_connmark_allowlist
 /* ipset.c */
 #ifdef HAVE_IPSET
 void ipset_init(void);
-int add_to_ipset(const char *setname, const union all_addr *ipaddr, int flags, int remove);
+int add_to_ipset(const char *setname, const union all_addr *ipaddr, int flags, int remove, int timeout);
 #endif
 
 /* pattern.c */
diff -purN dnsmasq-2.86.orig/src/ipset.c dnsmasq-2.86/src/ipset.c
--- dnsmasq-2.86.orig/src/ipset.c	2021-09-08 22:21:22.000000000 +0200
+++ dnsmasq-2.86/src/ipset.c	2022-07-17 08:41:02.778294613 +0200
@@ -36,6 +36,7 @@
 #define IPSET_ATTR_IPADDR_IPV6 2
 #define IPSET_ATTR_PROTOCOL 1
 #define IPSET_ATTR_SETNAME 2
+#define IPSET_ATTR_TIMEOUT 6
 #define IPSET_CMD_ADD 9
 #define IPSET_CMD_DEL 10
 #define IPSET_MAXNAMELEN 32
@@ -99,7 +100,7 @@ void ipset_init(void)
   die (_("failed to create IPset control socket: %s"), NULL, EC_MISC);
 }
 
-static int new_add_to_ipset(const char *setname, const union all_addr *ipaddr, int af, int remove)
+static int new_add_to_ipset(const char *setname, const union all_addr *ipaddr, int af, int remove, int timeout)
 {
   struct nlmsghdr *nlh;
   struct my_nfgenmsg *nfg;
@@ -132,6 +133,11 @@ static int new_add_to_ipset(const char *
   nested[0] = (struct my_nlattr *)(buffer + NL_ALIGN(nlh->nlmsg_len));
   nlh->nlmsg_len += NL_ALIGN(sizeof(struct my_nlattr));
   nested[0]->nla_type = NLA_F_NESTED | IPSET_ATTR_DATA;
+  if (!remove && timeout > 0)
+    {
+      uint32_t tm = htonl(timeout);
+      add_attr(nlh, IPSET_ATTR_TIMEOUT | NLA_F_NET_BYTEORDER, sizeof(tm), &tm);
+    }
   nested[1] = (struct my_nlattr *)(buffer + NL_ALIGN(nlh->nlmsg_len));
   nlh->nlmsg_len += NL_ALIGN(sizeof(struct my_nlattr));
   nested[1]->nla_type = NLA_F_NESTED | IPSET_ATTR_IP;
@@ -189,7 +195,7 @@ static int old_add_to_ipset(const char *
 
 
 
-int add_to_ipset(const char *setname, const union all_addr *ipaddr, int flags, int remove)
+int add_to_ipset(const char *setname, const union all_addr *ipaddr, int flags, int remove, int timeout)
 {
   int ret = 0, af = AF_INET;
 
@@ -205,7 +211,7 @@ int add_to_ipset(const char *setname, co
     }
   
   if (ret != -1) 
-    ret = old_kernel ? old_add_to_ipset(setname, ipaddr, remove) : new_add_to_ipset(setname, ipaddr, af, remove);
+    ret = old_kernel ? old_add_to_ipset(setname, ipaddr, remove) : new_add_to_ipset(setname, ipaddr, af, remove, timeout);
 
   if (ret == -1)
      my_syslog(LOG_ERR, _("failed to update ipset %s: %s"), setname, strerror(errno));
diff -purN dnsmasq-2.86.orig/src/rfc1035.c dnsmasq-2.86/src/rfc1035.c
--- dnsmasq-2.86.orig/src/rfc1035.c	2021-09-08 22:21:22.000000000 +0200
+++ dnsmasq-2.86/src/rfc1035.c	2022-07-17 08:50:28.436129610 +0200
@@ -845,7 +845,7 @@ int extract_addresses(struct dns_header
 		      while (*ipsets_cur)
 			{
 			  log_query((flags & (F_IPV4 | F_IPV6)) | F_IPSET, name, &addr, *ipsets_cur);
-			  add_to_ipset(*ipsets_cur++, &addr, flags, 0);
+			  add_to_ipset(*ipsets_cur++, &addr, flags, 0, attl);
 			}
 		    }
 #endif
