diff -purN ipppd-3.25+dfsg1.orig/ipppd/lcp.c ipppd-3.25+dfsg1/ipppd/lcp.c
--- ipppd-3.25+dfsg1.orig/ipppd/lcp.c	2012-07-06 08:27:19.000000000 +0200
+++ ipppd-3.25+dfsg1/ipppd/lcp.c	2017-11-25 21:26:52.874837479 +0100
@@ -51,10 +51,10 @@ char lcp_rcsid[] = "$Id: lcp.c,v 1.12 20
 
 #ifdef __linux__		/* Needs ppp ioctls */
 #if defined __GLIBC__ && __GLIBC__ >= 2
-# include </usr/include/net/if.h>
-# include </usr/include/net/if_ppp.h>
+# include <net/if.h>
+# include <net/if_ppp.h>
 #else
-# include </usr/include/net/if.h>
+# include <net/if.h>
 # include <linux/if_ppp.h>
 #endif
 #endif
diff -purN ipppd-3.25+dfsg1.orig/ipppd/main.c ipppd-3.25+dfsg1/ipppd/main.c
--- ipppd-3.25+dfsg1.orig/ipppd/main.c	2017-11-25 21:12:45.303972784 +0100
+++ ipppd-3.25+dfsg1/ipppd/main.c	2017-11-25 21:27:16.242476156 +0100
@@ -48,7 +48,7 @@ char main_rcsid[] = "$Id: main.c,v 1.26
 #include <sys/resource.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
-#include </usr/include/net/if.h>
+#include <net/if.h>
 
 #include "fsm.h"
 #include "ipppd.h"
diff -purN ipppd-3.25+dfsg1.orig/ipppd/sys-linux.c ipppd-3.25+dfsg1/ipppd/sys-linux.c
--- ipppd-3.25+dfsg1.orig/ipppd/sys-linux.c	2012-07-06 08:27:19.000000000 +0200
+++ ipppd-3.25+dfsg1/ipppd/sys-linux.c	2017-11-25 21:26:33.212141547 +0100
@@ -51,14 +51,14 @@ char sys_rcsid[] = "$Id: sys-linux.c,v 1
 #include <unistd.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-#include </usr/include/net/if.h>
-#include </usr/include/net/if_arp.h>
-#include </usr/include/net/route.h>
+#include <net/if.h>
+#include <net/if_arp.h>
+#include <net/route.h>
 #if defined __GLIBC__ && __GLIBC__ >= 2
-# include </usr/include/net/ppp_defs.h>
-# include </usr/include/net/if_ppp.h>
+# include <net/ppp_defs.h>
+# include <net/if_ppp.h>
 # include "ippp-filter-compat.h"
-# include </usr/include/net/ethernet.h>
+# include <net/ethernet.h>
 # include "route.h"
 #else
 # include <linux/ppp_defs.h>
