diff -urN isdnutils-3.25.orig/ipppd/main.c isdnutils-3.25/ipppd/main.c
--- isdnutils-3.25.orig/ipppd/main.c	2014-08-10 08:12:02.970934257 +0200
+++ isdnutils-3.25/ipppd/main.c	2014-08-10 08:20:19.410970516 +0200
@@ -1295,6 +1295,7 @@
     case 'I':
         str = ip_ntoa(va_arg(args, u_int32_t));
         break;
+#if 0
     case 'r':
         f = va_arg(args, char *);
         a = va_arg(args, va_list);
@@ -1306,6 +1307,7 @@
         buf += n;
         buflen -= n;
         continue;
+#endif
     case 't':
         time(&t);
         str = ctime(&t);
