diff -urN cron-3.0pl1.orig/Makefile cron-3.0pl1.patched/Makefile
--- cron-3.0pl1.orig/Makefile	2012-03-16 10:43:37.000000000 +0100
+++ cron-3.0pl1.patched/Makefile	2012-03-16 10:44:26.000000000 +0100
@@ -87,7 +87,7 @@
 #################################### end configurable stuff
 
 SHELL		=	/bin/sh
-CFLAGS		+=	$(INCLUDE) $(COMPAT) $(DEFS)
+CFLAGS		+=	$(INCLUDE) $(COMPAT) $(DEFS) $(OPTIM)
 
 INFOS		=	README CHANGES FEATURES INSTALL CONVERSION THANKS MAIL
 MANPAGES	=	bitstring.3 crontab.5 crontab.1 cron.8 putman.sh
