diff -urN xmlrpc-c-1.33.17.orig/lib/expat/gennmtab/Makefile xmlrpc-c-1.33.17/lib/expat/gennmtab/Makefile
--- xmlrpc-c-1.33.17.orig/lib/expat/gennmtab/Makefile	2015-01-23 03:57:50.000000000 +0100
+++ xmlrpc-c-1.33.17/lib/expat/gennmtab/Makefile	2015-03-29 18:19:11.754777687 +0200
@@ -38,9 +38,9 @@
 dep: dep-common
 
 gennmtab.o:%.o:%.c
-	$(BUILDTOOL_CC) -c $< -o $@ $(CFLAGS_ALL) $(INCLUDES)
+	$(CC_FOR_BUILD) -c $< -o $@ $(CFLAGS_FOR_BUILD) $(INCLUDES)
 
 gennmtab:%:%.o
-	$(BUILDTOOL_CCLD) -o $@ $(LDFLAGS_ALL) $^
+	$(CC_FOR_BUILD) -o $@ $(LDFLAGS_FOR_BUILD) $^
 
 include depend.mk
