diff -urN mgetty-1.1.37.orig/policy.h-dist mgetty-1.1.37/policy.h-dist
--- mgetty-1.1.37.orig/policy.h-dist	2010-06-05 11:57:00.000000000 +0200
+++ mgetty-1.1.37/policy.h-dist	2012-07-29 19:50:17.744812271 +0200
@@ -11,14 +11,14 @@
 
 /* main mgetty configuration file
  */
-#define MGETTY_CONFIG "mgetty.config"
+#define MGETTY_CONFIG "/etc/mgetty+sendfax/mgetty.config"
 
 /* sendfax configuration file
  *
  * if path doesn't start with "/", it's relative to CONFDIR (Makefile)
  * if not defined, no configuration file is read (saves a few kbytes)
  */
-#define SENDFAX_CONFIG "sendfax.config"
+#define SENDFAX_CONFIG "/etc/mgetty+sendfax/sendfax.config"
 
 
 /* login dispatcher config file (for mgetty)
@@ -39,7 +39,7 @@
  *
  * If this doesn't start with "/", it's relative to CONFDIR.
  */
-#define LOGIN_CFG_FILE "login.config"
+#define LOGIN_CFG_FILE "/etc/mgetty+sendfax/login.config"
 
 /* default login program
  *
@@ -65,7 +65,7 @@
  * Not needed if PPP callers want to get a real "login:"
  * prompt first. Don't forget to activate the /AutoPPP/ line in login.config!
  */
-/* #define AUTO_PPP */
+#define AUTO_PPP
 
 /* callback config file
  *
@@ -110,11 +110,11 @@
 /* user id of the "uucp" user. The tty device will be owned by this user,
  * so parallel dial-out of uucico will be possible
  */
-#define DEVICE_OWNER	"uucp"
+#define DEVICE_OWNER	"root"
 /* group id that the device is chown()ed to. If not defined, the
  * primary group of "DEVICE_OWNER" is used.
  */
-#define DEVICE_GROUP	"modem"
+#define DEVICE_GROUP	"root"
 
 /* access mode for the line while getty has it - it should be accessible
  * by uucp / uucp, but not by others (imagine someone dialing into your
@@ -370,7 +370,7 @@
 /* incoming faxes will be chown()ed to this uid and gid.
  * if FAX_IN_GROUP is undefined, the group of ...OWNER is used.
  */
-#define FAX_IN_OWNER	"uucp"
+#define FAX_IN_OWNER	"root"
 /* #define FAX_IN_GROUP	"root" */
 
 /* incoming faxes will be chmod()ed to this mode
@@ -586,7 +586,7 @@
  * If you don't want this type of service, do not define it at all
  * Absolute path name has to be used here!
  */
-#define FAX_NOTIFY_PROGRAM "/usr/local/lib/mgetty+sendfax/new_fax"
+#define FAX_NOTIFY_PROGRAM "/etc/mgetty+sendfax/new_fax"
 
 /* default minimum space required on spooling partition for receiving a FAX
  * (in KILObytes)
