diff -urN mgetty-1.1.37.orig/mgetty.c mgetty-1.1.37/mgetty.c
--- mgetty-1.1.37.orig/mgetty.c	2009-03-19 16:14:28.000000000 +0100
+++ mgetty-1.1.37/mgetty.c	2012-08-08 14:30:13.793062860 +0200
@@ -361,7 +361,6 @@
     lprintf( L_MESG, "vgetty: %s", vgetty_version);
 #endif
     lprintf( L_MESG, "mgetty: %s", mgetty_version);
-    lprintf( L_NOISE, "%s compiled at %s, %s", __FILE__, __DATE__, __TIME__ );
     i=getppid();
     lprintf( L_NOISE, "user id: %d, pid: %d, parent pid: %d", getuid(), getpid(), i);
     if ( i != 1 )
diff -urN mgetty-1.1.37.orig/sendfax.c mgetty-1.1.37/sendfax.c
--- mgetty-1.1.37.orig/sendfax.c	2009-03-19 16:13:43.000000000 +0100
+++ mgetty-1.1.37/sendfax.c	2012-08-08 14:30:22.205197300 +0200
@@ -322,7 +322,6 @@
     sendfax_get_config( NULL );
 
     lprintf( L_MESG, "sendfax: %s", mgetty_version );
-    lprintf( L_NOISE, "%s compiled at %s, %s", __FILE__, __DATE__, __TIME__ );
 
     /* for simplicity, put a few config things into global variables */
     verbose = c_bool( verbose );
