diff -urN syslinux-6.02.orig/Makefile syslinux-6.02/Makefile
--- syslinux-6.02.orig/Makefile	2013-10-13 19:59:03.000000000 +0200
+++ syslinux-6.02/Makefile	2015-03-03 22:01:46.455667279 +0100
@@ -191,7 +191,7 @@
 IOBJECTS = $(ITARGET) \
 	utils/gethostip utils/isohybrid utils/mkdiskimage \
 	mtools/syslinux linux/syslinux extlinux/extlinux
-ISUBDIRS = libinstaller mtools linux extlinux utils
+ISUBDIRS = libinstaller mtools cdemul linux extlinux utils
 
 # Things to install in /usr/bin
 INSTALL_BIN   =	mtools/syslinux
@@ -294,7 +294,7 @@
 
 #
 # List the dependencies to help out parallel builds.
-dos extlinux linux mtools win32 win64: libinstaller
+dos extlinux linux mtools cdemul win32 win64: libinstaller
 libinstaller: core
 utils: mbr
 core: com32
diff -urN syslinux-6.02.orig/cdemul/syslinux.c syslinux-6.02/cdemul/syslinux.c
--- syslinux-6.02.orig/cdemul/syslinux.c	2013-10-13 19:59:03.000000000 +0200
+++ syslinux-6.02/cdemul/syslinux.c	2015-03-03 22:02:54.560662867 +0100
@@ -276,10 +276,9 @@
 	    "MTOOLS_SKIP_CHECK=1\n"
 	    "MTOOLS_FAT_COMPATIBILITY=1\n"
 	    "drive s:\n"
-	    "  file=\"/proc/%lu/fd/%d\"\n"
+	    "  file=\"%s\"\n"
 	    "  offset=%llu\n",
-	    (unsigned long)mypid,
-	    dev_fd, (unsigned long long)opt.offset);
+	    opt.device, (unsigned long long)opt.offset);
 
     if (ferror(mtc) || fclose(mtc))
 	die_err(mtools_conf);
