diff -urN syslinux-6.03.orig/mk/embedded.mk syslinux-6.03/mk/embedded.mk
--- syslinux-6.03.orig/mk/embedded.mk	2014-10-06 18:27:44.000000000 +0200
+++ syslinux-6.03/mk/embedded.mk	2017-01-17 08:50:22.069208126 +0100
@@ -55,7 +55,7 @@
 
 LIBGCC    := $(shell $(CC) $(GCCOPT) --print-libgcc)
 
-LD        += -m elf_$(ARCH)
+override LD        += -m elf_$(ARCH)
 
 # Note: use += for CFLAGS and SFLAGS in case something is set in MCONFIG.local
 CFLAGS    += $(GCCOPT) -g $(GCCWARN) -Wno-sign-compare $(OPTFLAGS) $(INCLUDES)
