diff -purN syslinux-6.03.orig/extlinux/Makefile syslinux-6.03/extlinux/Makefile
--- syslinux-6.03.orig/extlinux/Makefile	2014-10-06 18:27:44.000000000 +0200
+++ syslinux-6.03/extlinux/Makefile	2018-09-11 22:13:07.679594708 +0200
@@ -19,7 +19,7 @@ include $(MAKEDIR)/syslinux.mk
 OPTFLAGS = -g -Os
 INCLUDES = -I$(SRC) -I$(objdir) -I$(SRC)/../libinstaller
 CFLAGS	 = $(CFLAGS_FOR_BUILD) $(GCCWARN) -Wno-sign-compare -D_FILE_OFFSET_BITS=64 \
-	   $(OPTFLAGS) $(INCLUDES)
+	   $(OPTFLAGS) $(INCLUDES) $(EXTRA_INCLUDES)
 LDFLAGS	 = $(LDFLAGS_FOR_BUILD)
 
 SRCS     = main.c \
