diff -urN horst-4.2.orig/display-help.c horst-4.2/display-help.c
--- horst-4.2.orig/display-help.c	2014-10-01 11:24:38.000000000 +0200
+++ horst-4.2/display-help.c	2015-03-29 14:05:17.610484434 +0200
@@ -39,7 +39,7 @@
 	print_centered(win, 2, COLS, "HORST - Horsts OLSR Radio Scanning Tool (or)");
 	print_centered(win, 3, COLS, "HORST - Highly Optimized Radio Scanning Tool");
 
-	print_centered(win, 5, COLS, "Version " VERSION " (build date " __DATE__ " " __TIME__ ")");
+	print_centered(win, 5, COLS, "Version " VERSION);
 	print_centered(win, 6, COLS, "(C) 2005-2014 Bruno Randolf, Licensed under the GPLv2");
 
 	mvwprintw(win, 8, 2, "Known IEEE802.11 Packet Types:");
