diff -urN horst-4.2.orig/Makefile horst-4.2/Makefile
--- horst-4.2.orig/Makefile	2014-10-01 11:24:38.000000000 +0200
+++ horst-4.2/Makefile	2015-03-29 14:46:27.868658624 +0200
@@ -29,14 +29,15 @@
 	display-spectrum.o display-channel.o control.o \
 	radiotap/radiotap.o
 LIBS=-lncurses -lm
-CFLAGS+=-Wall -Wextra -g -I.
+CFLAGS+=-Wall -Wextra -g
+override CFLAGS+=-I.
 
 ifeq ($(DEBUG),1)
-CFLAGS+=-DDO_DEBUG
+override CFLAGS+=-DDO_DEBUG
 endif
 
 ifeq ($(PCAP),1)
-CFLAGS+=-DPCAP
+override CFLAGS+=-DPCAP
 LIBS+=-lpcap
 endif
 
