diff -purN iptables-1.6.1.orig/iptables/ip6tables.c iptables-1.6.1/iptables/ip6tables.c
--- iptables-1.6.1.orig/iptables/ip6tables.c	2017-01-26 17:11:58.000000000 +0100
+++ iptables-1.6.1/iptables/ip6tables.c	2018-02-15 14:34:17.692427377 +0100
@@ -1338,7 +1338,7 @@ int do_command6(int argc, char *argv[],
 	struct in6_addr *smasks = NULL, *dmasks = NULL;
 
 	int verbose = 0;
-	int wait = 0;
+	int wait = -1;
 	struct timeval wait_interval = {
 		.tv_sec	= 1,
 	};
diff -purN iptables-1.6.1.orig/iptables/iptables.c iptables-1.6.1/iptables/iptables.c
--- iptables-1.6.1.orig/iptables/iptables.c	2017-01-26 17:11:58.000000000 +0100
+++ iptables-1.6.1/iptables/iptables.c	2018-02-15 14:33:57.236743743 +0100
@@ -1336,7 +1336,7 @@ int do_command4(int argc, char *argv[],
 	};
 	bool wait_interval_set = false;
 	int verbose = 0;
-	int wait = 0;
+	int wait = -1;
 	const char *chain = NULL;
 	const char *shostnetworkmask = NULL, *dhostnetworkmask = NULL;
 	const char *policy = NULL, *newname = NULL;
