diff -urN sff-1.0/Makefile sff-1.0.patched/Makefile
--- sff-1.0/Makefile	1998-10-09 18:14:07.000000000 +0200
+++ sff-1.0.patched/Makefile	2012-03-21 00:05:16.000000000 +0100
@@ -1,12 +1,12 @@
-CFLAGS=-Wall -I./jpeg-6a
-LDFLAGS=-L./jpeg-6a
+CFLAGS=-Wall
+LDFLAGS=
 CC=gcc
 LD=gcc
 
 prefix=/usr
 bindir=$(prefix)/bin
 
-sff2misc: jpeg-6a/libjpeg.a sff2misc.o sfffunc.o
+sff2misc: sff2misc.o sfffunc.o
 	$(LD) $(LDFLAGS) -o $@ $^ -ljpeg
 
 sff2misc.o: sff2misc.c sff.h
