diff -urN sff-1.0/sff.h sff-1.0.patched/sff.h
--- sff-1.0/sff.h	1998-10-09 17:47:37.000000000 +0200
+++ sff-1.0.patched/sff.h	2012-03-21 00:11:23.000000000 +0100
@@ -19,8 +19,8 @@
 
 typedef struct sff_dochdr {
 	__u32 magic         __attribute__ ((packed));
-	__u8  version       __attribute__ ((packed));
-	__u8  reserved      __attribute__ ((packed));
+	__u8  version;
+	__u8  reserved;
 	__u16 userinfo      __attribute__ ((packed));
 	__u16 pagecount     __attribute__ ((packed));
 	__u16 offsetfirstph __attribute__ ((packed));
@@ -31,10 +31,10 @@
 typedef struct sff_paghdr {
 //	__u8  id           __attribute__ ((packed));
 //	__u8  len          __attribute__ ((packed));
-	__u8  vres         __attribute__ ((packed));
-	__u8  hres         __attribute__ ((packed));
-	__u8  coding       __attribute__ ((packed));
-	__u8  reserverd    __attribute__ ((packed));
+	__u8  vres;
+	__u8  hres;
+	__u8  coding;
+	__u8  reserverd;
 	__u16 linelen      __attribute__ ((packed));
 	__u16 pagelen      __attribute__ ((packed));
 	__u32 offsetprev   __attribute__ ((packed));
