diff -urN libevdev-1.4.orig/Makefile.in libevdev-1.4/Makefile.in
--- libevdev-1.4.orig/Makefile.in	2015-03-03 23:50:37.000000000 +0100
+++ libevdev-1.4/Makefile.in	2015-03-29 21:55:29.592709551 +0200
@@ -371,7 +371,7 @@
 PRINT_DIRECTORY_FLAGS_0 = --no-print-directory
 PRINT_DIRECTORY_FLAGS_ = $(PRINT_DIRECTORY_FLAGS_$(AM_DEFAULT_VERBOSITY))
 AM_MAKEFLAGS = $(PRINT_DIRECTORY_FLAGS_$(V))
-SUBDIRS = doc libevdev tools test
+SUBDIRS = doc libevdev tools
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libevdev.pc
 EXTRA_DIST = libevdev.pc.in
