diff -urN libgcrypt-1.5.3.orig/compat/compat.c libgcrypt-1.5.3/compat/compat.c
--- libgcrypt-1.5.3.orig/compat/compat.c	2013-05-22 18:02:54.000000000 +0200
+++ libgcrypt-1.5.3/compat/compat.c	2013-11-20 20:23:43.388470641 +0100
@@ -32,7 +32,7 @@
     "          2010, 2011, 2012 Free Software Foundation, Inc.\n"
     "Copyright 2012, 2013 g10 Code GmbH\n"
     "\n"
-    "(" BUILD_REVISION " " BUILD_TIMESTAMP ")\n"
+    "(" BUILD_REVISION ")\n"
     "\n\n";
   return blurb;
 }
